---
title: "Transcriptomic analysis of potato samples infected with P. infestans - DE analysis"
author: "Liam Cremona, Niall Dunne, Ossama Edbali - Patricia Ponce"
date: "2022-2023"
output:
  html_document:
    self_contained: false
fontsize: 15pt
editor_options:
  markdown: 
    wrap: 72
---

> This script has been apadted from Patricia Ponce

# Introduction

The data comes from potato leaves infected with Phythophtora infestans
for 48 hours using a detached infection assay. Before this analysis the
reads were mapped to the potato genome v6 with STAR, then counted with
HTSeq. Three potato varieties were infected: Sarpo Shona (Tolerant),
Sarpo Mira (Tolerant) and Duke of York (Susceptible).

## Purpose

This part of the transcriptomic analysis involve the
normalization of the counts by DESeq2. After that we plot a PCA to
visualize the distribution of the samples, then find which genes are
significantly different between varieties 48 hours after infection.
All the comparisons will be done with against the mock (Infected-Area vs
Mock or Out-infection-Area vs Mock).

## Input

- HTseq counts in the `quantified` directory.
- Annotation info from the potato genome v6.1 in "Z:/1.PotatoDroughtOut_genomev6/4.GO_analysis/gProfile_FileHC.GeneTransID.Name.GO/potv6_Ath_potv404.RData".

## Output

- "1.DESeq2_analysis_blight_infection48h.html" (After kniting)
- "1.DESeq2_analysis_blight_infection48h_files" --\> Inside the folder
"figure-html" are all the graphs produced with this code. (After
kniting) -All the files in the "2.DESeq2_output_48h" folder

# Setup

Path of the folder for the inputs and outputs:

```{r}
setwd("/rds/projects/l/leachlj-potato-qtl-project/M6_Area/potato-blight-rna-seq-pipeline")
```

Packages to employ for the analysis:

```{r message=FALSE, warning=FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("apeglm")
```


```{r message=FALSE, warning=FALSE}
#this code works to install an older version of lasso2 from the archives
packageurl <- "https://cran.r-project.org/src/contrib/Archive/lasso2/lasso2_1.2-22.tar.gz"
install.packages(packageurl, repos=NULL, type="source")
```

```{r message=FALSE, warning=FALSE}
BiocManager::install("DEGreport")
```

```{r}
BiocManager::install("EnhancedVolcano")
```

```{r}
library(apeglm)
library(DESeq2)
library(DEGreport)
library(EnhancedVolcano)
library(ggplot2)
library(pheatmap)
library(RColorBrewer)
library(pheatmap)
library(dplyr) 
library(tibble)
library(data.table)
library(gridExtra)
library(stringr)
library(VennDiagram)
```

```{r}
source('./functions.R')
source('./comparisons.R')
```

```{r}
#specify this to enable plotting to work
options(bitmapType='cairo')
```

# Input configuration

```{r}
use_v4 = FALSE
```

# Prepare data

```{r}
directory = './quantified'
sampleFiles = grep('*count.txt', list.files(directory), value = TRUE)
sampleFilesR = sort(sampleFiles[startsWith(sampleFiles, 'R')])
sampleFilesA = sort(sampleFiles[startsWith(sampleFiles, 'A')])
sampleFiles = c(sampleFilesR, sampleFilesA)
sampleNames = c(
  'SMira_C1', 'SMira_C2', 'SMira_C3',
  'SMira_OI1', 'SMira_OI2', 'SMira_OI3',
  'Duke_C1', 'Duke_C2', 'Duke_C3',
  'Duke_OI1', 'Duke_OI2', 'Duke_OI3',
  'SShona_C1', 'SShona_C2', 'SShona_C3',
  'SShona_OI1', 'SShona_OI2', 'SShona_OI3',
  'SMira_I1', 'SMira_I2', 'SMira_I3',
  'Duke_I1', 'Duke_I2', 'Duke_I3',
  'SShona_I1', 'SShona_I2', 'SShona_I3',
  'SMira_M1', 'SMira_M2', 'SMira_M3',
  'Duke_M1', 'Duke_M2', 'Duke_M3',
  'SShona_M1', 'SShona_M2', 'SShona_M3'
)

sampleReplicate = rep(c(1, 2, 3), 12)

sampleVariety = c(
  rep('SMira', 6),
  rep('Duke', 6),
  rep('SShona', 6),
  rep('SMira', 3),
  rep('Duke', 3),
  rep('SShona', 3),
  rep('SMira', 3),
  rep('Duke', 3),
  rep('SShona', 3)
)

sampleTreatment = c(
  rep('Control', 3),
  rep('OutInf', 3),
  rep('Control', 3),
  rep('OutInf', 3),
  rep('Control', 3),
  rep('OutInf', 3),
  rep('Inf', 9),
  rep('Mock', 9)
)

sampleTable = data.frame(sampleNames= sampleNames, 
                          fileName = sampleFiles, 
                          replicate = sampleReplicate,
                          variety = sampleVariety,
                          treatment = sampleTreatment)

sampleTable$replicate = factor(sampleTable$replicate)
sampleTable$variety = factor(sampleTable$variety)
sampleTable$treatment = factor(sampleTable$treatment)

varieties <- list(
  Duke=list(
    type="susceptible"
  ),
  SMira=list(
    type="tolerant"
  ),
  SShona=list(
    type="tolerant"
  )
)
```

# Run DESeq2 with all the samples

```{r}
dds_Blight <- DESeqDataSetFromHTSeqCount(
  sampleTable = sampleTable,
  directory = directory,
  design = ~variety + treatment + variety:treatment
)

# Removing genes whose counts is less than 10 in total around all the libraries
keep <- rowSums(counts(dds_Blight)) >= 10
dds_Blight <- dds_Blight[keep,]

dds_Blight$variety <- relevel(dds_Blight$variety, ref = c("Duke"))
dds_Blight$treatment <- relevel(dds_Blight$treatment, ref = c("Mock"))
#dds_Blight$treatment <- relevel(dds_Blight$treatment, ref = c("Control"))

dds_Blight <- DESeq(dds_Blight)
resultsNames(dds_Blight) # it's the input for "name" and "contrast"
```

# Exploratory analysis

## Normalized counts

```{r}
Norm.Counts <- counts(dds_Blight, normalized=TRUE)
```

## Gene count distribution per samples in leaf

Distribution before normalization:

```{r}
boxplot(log2(counts(dds_Blight) + 1), range=0, las=2)
```

Distribution after normalization:

```{r}
boxplot(log2(counts(dds_Blight, normalized=TRUE) + 1), range=0, las=2)
```

## Principal Component Analysis (PCA) of the leaf samples

Variance stabilization transformation:

```{r}
vst_Blight <- vst(dds_Blight, blind=TRUE)
```

PCA of all leaf samples:

```{r, echo=FALSE, fig.height=7, fig.width=14}
# n = 29393
pcaVST <- plotPCA(vst_Blight, intgroup = c("treatment","variety"), returnData = TRUE, ntop = 29393)
percentVar <- round(100*attr(pcaVST, "percentVar"))

SusTol_all <- qplot(PC1, PC2, color=treatment, shape=variety, data=pcaVST) +
  geom_point(size=3)+
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))+
  labs(title="All leaf samples_n29393")+
  geom_point()+
  geom_text(aes(label=colnames(vst_Blight)),size = 3, vjust=1.5, hjust=1)


# n = 1000
pcaVST_n1000 <- plotPCA(vst_Blight, intgroup = c("treatment","variety"), returnData = TRUE, ntop = 1000)
percentVar_n1000 <- round(100*attr(pcaVST_n1000, "percentVar"))

SusTol_n1000 <- qplot(PC1, PC2, color=treatment, shape=variety, data=pcaVST_n1000) +
           geom_point(size=3)+
           xlab(paste0("PC1: ",percentVar_n1000[1],"% variance")) +
           ylab(paste0("PC2: ",percentVar_n1000[2],"% variance"))+
           labs(title="All leaf samples_n1000")+
           geom_point()+
           geom_text(aes(label=colnames(vst_Blight)),size = 3, vjust=1.5, hjust=1)

grid.arrange(SusTol_all, SusTol_n1000,
             ncol=2, widths=unit(c(160,160), c("mm","mm")), heights=unit(150, "mm"))
```

The distribution of the samples is very similar in both cases, by using
all the normalized genes or using only the 1000 most variable genes. All
replicates looks good, then it won't be necessary to remove any sample.

## Heatmap and clustering of the samples

Heatmap of all leaf samples, after removing bad biological replicates:

```{r}
sampleDists <- dist(t(assay(vst_Blight)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vst_Blight$variety, vst_Blight$treatment, sep="-")
colnames(sampleDistMatrix) <- NULL

colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)

pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)
```

# Pairwise comparisons

> Tutorial: <https://rstudio-pubs-static.s3.amazonaws.com/329027_593046fb6d7a427da6b2c538caf601e1.html>

Data to compare:

```{r}
resultsNames(dds_Blight) # it's the input for "name" and "contrast"
```

For the comparison: - Base_level: Duke / Mock - res: To get the pvalue
and lfcThreshold=0, - default lfcThreshold=0

For the DE analysis, genes with the absolute value of the log2folchange
more than 1 (\|log2FC\| \> 1) will be selected.

Pairwise comparisons between samples (Inf/OutInf vs Mock and tolerant vs Duke in Infect/OutInfect. area) from the tolerant variety:

```{r}
comparisons = list(
  res.Duke.Out = c("treatment_OutInf_vs_Mock"),
  res.Duke.Inf = c("treatment_Inf_vs_Mock"),
  res.SMira.Out = c("treatment_OutInf_vs_Mock","varietySMira.treatmentOutInf"),
  res.SMira.Inf = c("treatment_Inf_vs_Mock","varietySMira.treatmentInf"),
  res.SShona.Out = c("treatment_OutInf_vs_Mock","varietySShona.treatmentOutInf"),
  res.SShona.Inf = c("treatment_Inf_vs_Mock","varietySShona.treatmentInf"),
  
  res.SMira_vs_Duke.Out = c("variety_SMira_vs_Duke" ,"varietySMira.treatmentOutInf"),
  res.SMira_vs_Duke.Inf = c("variety_SMira_vs_Duke" ,"varietySMira.treatmentInf"),
  res.SShona_vs_Duke.Out = c("variety_SShona_vs_Duke","varietySShona.treatmentOutInf"),
  res.SShona_vs_Duke.Inf = c("variety_SShona_vs_Duke","varietySShona.treatmentInf")
)

res <- create_comparisons(dds_Blight, comparisons)
```

# Volcano plots

This plot allows as to observe the ranges of the foldchanges and
visualize if a high or low amount of genes pass the threshold of
\|log2FC\| \> 1 & padj \< 0.05. In this case, the foldchange was
calculated from the OutINf/Mock and INf/Mock in each variety.

In all the samples:

```{r, echo=FALSE, fig.height=20, fig.width=14}
volcano_plots_from_comparisons(res, function(name) { !grepl("_vs_", name) })
```

# Selecting Differentially Expressed Genes (DEGs)

```{r}
degs = create_degs(res)
```

# Get up and down regulated genes

```{r}
up_down_degs = create_up_down_degs(degs)
```

# Venn diagrams

(1) Venn diagrams of DEG in each variety, Outinfection and Infection:

```{r, echo=FALSE, fig.height=8, fig.width=13}
#DEGs: |FC|>1, padj. <0.05:...........................................................................................................................................
Venn.Duke.DEG.UP <- venn_diagram_2way(
  list(Duke_Out=DEG.Duke.Out.UP$gene, Duke_Inf=DEG.Duke.Inf.UP$gene),
  "Duke of York - Up"
)

Venn.Duke.DEG.DOWN <- venn_diagram_2way(
  list(Duke_Out=DEG.Duke.Out.DOWN$gene, Duke_Inf=DEG.Duke.Inf.DOWN$gene),
  "Duke of York - Down"
)

Venn.SMira.DEG.UP <- venn_diagram_2way(
  list(S.Mira_Out=DEG.SMira.Out.UP$gene, S.Mira_Inf=DEG.SMira.Inf.UP$gene),
  "Sarpo Mira - Up"
)

Venn.SMira.DEG.DOWN <- venn_diagram_2way(
  list(S.Mira_Out=DEG.SMira.Out.DOWN$gene, S.Mira_Inf=DEG.SMira.Inf.DOWN$gene),
  "Sarpo Mira - Down"
)

Venn.SShona.DEG.UP <- venn_diagram_2way(
  list(S.Shona_Out=DEG.SShona.Out.UP$gene, S.Shona_Inf=DEG.SShona.Inf.UP$gene),
  "Sarpo Shona - Up"
)

Venn.SShona.DEG.DOWN <- venn_diagram_2way(
  list(S.Shona_Out=DEG.SShona.Out.DOWN$gene, S.Shona_Inf=DEG.SShona.Inf.DOWN$gene),
  "Sarpo Shona - Down"
)

#DEGs for only padj. <0.05, not selected by log2FC:...........................................................................................................................
Venn.Duke.DEG_p.UP <- venn_diagram_2way(
  list(Duke_Out=DEG_p.Duke.Out.UP$gene, Duke_Inf=DEG_p.Duke.Inf.UP$gene),
  "Duke of York - Up"
)

Venn.Duke.DEG_p.DOWN <- venn_diagram_2way(
  list(Duke_Out=DEG_p.Duke.Out.DOWN$gene, Duke_Inf=DEG_p.Duke.Inf.DOWN$gene),
  "Duke of York - Down"
)

Venn.SMira.DEG_p.UP <- venn_diagram_2way(
  list(S.Mira_Out=DEG_p.SMira.Out.UP$gene, S.Mira_Inf=DEG_p.SMira.Inf.UP$gene),
  "Sarpo Mira - Up"
)

Venn.SMira.DEG_p.DOWN <- venn_diagram_2way(
  list(S.Mira_Out=DEG_p.SMira.Out.DOWN$gene, S.Mira_Inf=DEG_p.SMira.Inf.DOWN$gene),
  "Sarpo Mira - Down"
)

Venn.SShona.DEG_p.UP <- venn_diagram_2way(
  list(S.Shona_Out=DEG_p.SShona.Out.UP$gene, S.Shona_Inf=DEG_p.SShona.Inf.UP$gene),
  "Sarpo Shona - Up"
)

Venn.SShona.DEG_p.DOWN <- venn_diagram_2way(
  list(S.Shona_Out=DEG_p.SShona.Out.DOWN$gene, S.Shona_Inf=DEG_p.SShona.Inf.DOWN$gene),
  "Sarpo Shona - Down"
)

grid.arrange(gTree(children = Venn.Duke.DEG.UP),  gTree(children = Venn.Duke.DEG.DOWN), 
             gTree(children = Venn.Duke.DEG_p.UP),  gTree(children = Venn.Duke.DEG_p.DOWN),
             gTree(children = Venn.SMira.DEG.UP), gTree(children = Venn.SMira.DEG.DOWN),
             gTree(children = Venn.SMira.DEG_p.UP), gTree(children = Venn.SMira.DEG_p.DOWN),
             gTree(children = Venn.SShona.DEG.UP),gTree(children = Venn.SShona.DEG.DOWN),
             gTree(children = Venn.SShona.DEG_p.UP),gTree(children = Venn.SShona.DEG_p.DOWN),
             top = "Nro. of Genes with a padj.<0.05 & |log2FC| > 1  -  Nro. of Genes with a padj.<0.05", ncol = 4, nrow = 3, 
             widths =unit(c(58,58,58,58), c("mm","mm","mm","mm")), 
             heights = unit(c(58,58,58), c("mm","mm","mm")))
```

(2) Venn diagrams of DEG in each condition:

```{r, echo=FALSE, fig.height=8, fig.width=13}
#DEGs (|FC|>1, padj. <0.05)
Venn.OutInf.DEG.UP <- venn_diagram_3way(
  list(Duke_Out=DEG.Duke.Out.UP$gene, S.Mira_Out=DEG.SMira.Out.UP$gene, S.Shona_Out=DEG.SShona.Out.UP$gene),
  "Out of Infection - Up"
)
             
Venn.OutInf.DEG.DOWN <- venn_diagram_3way(
  list(Duke_Out=DEG.Duke.Out.DOWN$gene, S.Mira_Out=DEG.SMira.Out.DOWN$gene, S.Shona_Out=DEG.SShona.Out.DOWN$gene),
  "Out of Infection - Down"
)
             
Venn.Inf.DEG.UP <- venn_diagram_3way(
  list(Duke_Inf=DEG.Duke.Inf.UP$gene, S.Mira_Inf=DEG.SMira.Inf.UP$gene, S.Shona_Inf=DEG.SShona.Inf.UP$gene),
  "Infection - Up"
)

Venn.Inf.DEG.DOWN <- venn_diagram_3way(
  list(Duke_Inf=DEG.Duke.Inf.DOWN$gene, S.Mira_Inf=DEG.SMira.Inf.DOWN$gene, S.Shona_Inf=DEG.SShona.Inf.DOWN$gene),
  "Infection - Down"
)

#DEGs but only seleceted by padj. <0.05:
Venn.OutInf.DEG_p.UP <- venn_diagram_3way(
  list(Duke_Out=DEG_p.Duke.Out.UP$gene, SMira_Out=DEG_p.SMira.Out.UP$gene, SShona_Out=DEG_p.SShona.Out.UP$gene),
  "Out of Infection - Up - any log2FC"
)

Venn.OutInf.DEG_p.DOWN <- venn_diagram_3way(
  list(Duke_Out=DEG_p.Duke.Out.DOWN$gene, SMira_Out=DEG_p.SMira.Out.DOWN$gene, SShona_Out=DEG_p.SShona.Out.DOWN$gene),
  "Out of Infection - Down - any log2FC"
)

Venn.Inf.DEG_p.UP <- venn_diagram_3way(
  list(Duke_Inf=DEG_p.Duke.Inf.UP$gene, SMira_Inf=DEG_p.SMira.Inf.UP$gene, SShona_Inf=DEG_p.SShona.Inf.UP$gene),
  "Infection - Up - any log2FC"
)

Venn.Inf.DEG_p.DOWN <- venn_diagram_3way(
  list(Duke_Inf=DEG_p.Duke.Inf.DOWN$gene, SMira_Inf=DEG_p.SMira.Inf.DOWN$gene, SShona_Inf=DEG_p.SShona.Inf.DOWN$gene),
  "Infection - Down - any log2FC"
)

grid.arrange(gTree(children = Venn.OutInf.DEG.UP),  gTree(children = Venn.OutInf.DEG.DOWN),
             gTree(children = Venn.OutInf.DEG_p.UP),  gTree(children = Venn.OutInf.DEG_p.DOWN), 
             gTree(children = Venn.Inf.DEG.UP), gTree(children = Venn.Inf.DEG.DOWN), 
             gTree(children = Venn.Inf.DEG_p.UP), gTree(children = Venn.Inf.DEG_p.DOWN),
             top = "Nro. of Genes with a padj.<0.05 & |log2FC| > 1      -      Nro. of Genes with a padj.<0.05",
             ncol = 4, nrow = 2, widths =unit(c(80,80,80,80), c("mm","mm","mm","mm")), heights = unit(c(80,80), c("mm","mm")))
```

# Merging DEGs with their GO and functional annotations

```{r}
if (use_v4) {
  load(file = "./analysis-data/potv6_Ath_potv404.RData")
  colnames(potv6_Ath_potv404)[3] <- "gene"
} else {
  go_annotations = read.table(file = './analysis-data/DM_1-3_516_R44_potato.v6.1.working_models.go.txt', sep = '\t', header = FALSE)
  go_annotations = go_annotations %>%
    mutate(gene = str_remove(V2, '\\.\\d+$')) %>%
    select(V2, gene, V5, V6) %>%
    rename(transcript_id = V2, GOterm = V5, TAIR = V6)
  
  func_annotations = read.table(file = './analysis-data/DM_1-3_516_R44_potato.v6.1.working_models.func_anno.txt', sep = '\t', header = FALSE)
  func_annotations = func_annotations %>%
    mutate(gene = str_remove(V1, '\\.\\d+$')) %>%
    rename(transcript_id = V1, name = V2)
}
```

```{r}
go_and_func_enrichment <- function (.data) {
  if (use_v4) {
    merge(.data, potv6_Ath_potv404, by ="gene", all.x = T)
  } else {
    merge(merge(.data, go_annotations, by="gene", all.x = T), func_annotations, by="transcript_id", all.x = T) %>%
      select(-gene.y) %>%
      rename(gene = gene.x)
  }
}
```

Adding annotation names to the result of the comparisons:

```{r}
annotated_degs <- lapply(degs, go_and_func_enrichment)
annotated_res <- lapply(res, function(x) { go_and_func_enrichment(x %>% to_tibble()) })
```

# Selecting DEG and DEG_p that are only in the tolerant but not in the susceptible after 48h of infection

(1) Selecting DEGs in the intersection of TOLs but not in Duke, with annotation(\|log2FC\| \> 1 & padj. \< 0.05)

(1.1) Process tolerant varieties in out of infected areas

```{r}
# indicesToRemove <- c(1, 3, 5, 6, 7)
# c(2,4,5,6,9) -> potv4
only_tol_degs <- only_in_tolerant(degs, up_down_degs, varieties)

indices_to_remove <- c("transcript_id", "baseMean", "lfcSE", "stat", "pvalue")
merge_suffixes <- c('_SMira', '_SShona')

OnlyTol.Out.DEG.up.genes <- only_in_a_variety_type(DEG.SMira.Out.UP$gene, DEG.SShona.Out.UP$gene, DEG.Duke.Out.UP$gene)
OnlyTol.Out.DEG.down.genes <- only_in_a_variety_type(DEG.SMira.Out.DOWN$gene, DEG.SShona.Out.DOWN$gene, DEG.Duke.Out.DOWN$gene)

OnlyTol.Out.DEG.up.annotation <- merge_with_annotations(
  OnlyTol.Out.DEG.up.genes,
  DEG.SMira.Out.annotation,
  DEG.SShona.Out.annotation,
  indices_to_remove,
  merge_suffixes)

OnlyTol.Out.DEG.down.annotation <- merge_with_annotations(
  OnlyTol.Out.DEG.down.genes,
  DEG.SMira.Out.annotation,
  DEG.SShona.Out.annotation,
  indices_to_remove,
  merge_suffixes)
```

(1.2) Process tolerant varieties in infected areas

```{r}
OnlyTol.Inf.DEG.up.genes <- only_in_a_variety_type(DEG.SMira.Inf.UP$gene, DEG.SShona.Inf.UP$gene, DEG.Duke.Inf.UP$gene)
OnlyTol.Inf.DEG.down.genes <- only_in_a_variety_type(DEG.SMira.Inf.DOWN$gene, DEG.SShona.Inf.DOWN$gene, DEG.Duke.Inf.DOWN$gene)

OnlyTol.Inf.DEG.up.annotation <- merge_with_annotations(
  OnlyTol.Inf.DEG.up.genes,
  DEG.SMira.Inf.annotation,
  DEG.SShona.Inf.annotation,
  indices_to_remove,
  merge_suffixes)

OnlyTol.Inf.DEG.down.annotation <- merge_with_annotations(
  OnlyTol.Inf.DEG.down.genes,
  DEG.SMira.Inf.annotation,
  DEG.SShona.Inf.annotation,
  indices_to_remove,
  merge_suffixes)
```

(2)Selecting DEG (\|log2FC\| \> 1 & padj. \< 0.05) not adding
annotation:

```{r}
#Intersect Sarpo Mira & Sarpo Shona but not Duke
View(as.data.frame(setdiff(intersect(DEG.SMira.Inf.UP$gene,DEG.SShona.Inf.UP$gene),DEG.Duke.Inf.UP$gene)))
View(as.data.frame(setdiff(intersect(DEG.SMira.Inf.DOWN$gene,DEG.SShona.Inf.DOWN$gene),DEG.Duke.Inf.DOWN$gene)))

#DEG only Sarpo Mira
View(as.data.frame(setdiff(setdiff(DEG.SMira.Inf.UP$gene,DEG.SShona.Inf.UP$gene),DEG.Duke.Inf.UP$gene)))
View(as.data.frame(setdiff(setdiff(DEG.SMira.Inf.DOWN$gene,DEG.SShona.Inf.DOWN$gene),DEG.Duke.Inf.DOWN$gene)))

#DEG only Sarpo Shona
View(as.data.frame(setdiff(setdiff(DEG.SShona.Inf.UP$gene,DEG.SMira.Inf.UP$gene),DEG.Duke.Inf.UP$gene)))
View(as.data.frame(setdiff(setdiff(DEG.SShona.Inf.DOWN$gene,DEG.SMira.Inf.DOWN$gene),DEG.Duke.Inf.DOWN$gene)))
```

(3)Selecting DEG_p (& padj. \< 0.05)

```{r}
OnlyTol.Out.DEG_p.up.genes <- only_in_a_variety_type(DEG_p.SMira.Out.UP$gene, DEG_p.SShona.Out.UP$gene, DEG_p.Duke.Out.UP$gene)
OnlyTol.Out.DEG_p.down.genes <- only_in_a_variety_type(DEG_p.SMira.Out.DOWN$gene, DEG_p.SShona.Out.DOWN$gene, DEG_p.Duke.Out.DOWN$gene)

OnlyTol.Out.DEG_p.up.annotation <- merge_with_annotations(
  OnlyTol.Out.DEG_p.up.genes,
  DEG_p.SMira.Out.annotation,
  DEG_p.SShona.Out.annotation,
  indices_to_remove,
  suffixes=c("_SMira", "_SShona"))

OnlyTol.Out.DEG_p.down.annotation <- merge_with_annotations(
  OnlyTol.Out.DEG_p.down.genes,
  DEG_p.SMira.Out.annotation,
  DEG_p.SShona.Out.annotation,
  indices_to_remove,
  suffixes=c("_SMira", "_SShona"))

#Gene list in "area of infection region" in both varieties
OnlyTol.Inf.DEG_p.up.genes <- only_in_a_variety_type(DEG_p.SMira.Inf.UP$gene, DEG_p.SShona.Inf.UP$gene, DEG_p.Duke.Inf.UP$gene)
OnlyTol.Inf.DEG_p.down.genes <- only_in_a_variety_type(DEG_p.SMira.Inf.DOWN$gene, DEG_p.SShona.Inf.DOWN$gene, DEG_p.Duke.Inf.DOWN$gene)

OnlyTol.Inf.DEG_p.up.annotation <- merge_with_annotations(
  OnlyTol.Inf.DEG_p.up.genes,
  DEG_p.SMira.Inf.annotation,
  DEG_p.SShona.Inf.annotation,
  indices_to_remove,
  suffixes=c("_SMira", "_SShona"))

OnlyTol.Inf.DEG_p.down.annotation <- merge_with_annotations(
  OnlyTol.Inf.DEG_p.down.genes,
  DEG_p.SMira.Inf.annotation,
  DEG_p.SShona.Inf.annotation,
  indices_to_remove,
  suffixes=c("_SMira", "_SShona"))
```

# Selecting DEG in the intersection Inf48 & Out48, in each variety:

```{r}
SMira.Inf_Out.DEG.up.genes <- in_intersection(DEG.SMira.Inf.UP$gene, DEG.SMira.Out.UP$gene)

SShona.Inf_Out.DEG.up.genes <- in_intersection(DEG.SShona.Inf.UP$gene, DEG.SShona.Out.UP$gene)

Duke.Inf_Out.DEG.up.genes <- in_intersection(DEG.Duke.Inf.UP$gene, DEG.Duke.Out.UP$gene)

SMira.Inf_Out.DEG.DOWN.genes <- in_intersection(DEG.SMira.Inf.DOWN$gene, DEG.SMira.Out.DOWN$gene)

SShona.Inf_Out.DEG.DOWN.genes <- in_intersection(DEG.SShona.Inf.DOWN$gene, DEG.SShona.Out.DOWN$gene)

Duke.Inf_Out.DEG.DOWN.genes <- in_intersection(DEG.Duke.Inf.DOWN$gene, DEG.Duke.Out.DOWN$gene)

# Adding annotations
SMira.Inf_Out.DEG.up.exp <- merge_with_annotations(
  SMira.Inf_Out.DEG.up.genes,
  DEG.SMira.Inf.annotation,
  DEG.SMira.Out.annotation,
  indices_to_remove,
  suffixes=c('_SMira_Inf', '_SMira_Out'))

SShona.Inf_Out.DEG.up.exp <- merge_with_annotations(
  SShona.Inf_Out.DEG.up.genes,
  DEG.SShona.Inf.annotation,
  DEG.SShona.Out.annotation,
  indices_to_remove,
  suffixes=c('_SShona_Inf', '_SShona_Out'))

Duke.Inf_Out.DEG.up.exp <- merge_with_annotations(
  Duke.Inf_Out.DEG.up.genes,
  DEG.Duke.Inf.annotation,
  DEG.Duke.Out.annotation,
  indices_to_remove,
  suffixes=c('_Duke_Inf', '_Duke_Out'))

SMira.Inf_Out.DEG.DOWN.exp <- merge_with_annotations(
  SMira.Inf_Out.DEG.DOWN.genes,
  DEG.SMira.Inf.annotation,
  DEG.SMira.Out.annotation,
  indices_to_remove,
  suffixes=c('_SMira_Inf', '_SMira_Out'))

SShona.Inf_Out.DEG.DOWN.exp <- merge_with_annotations(
  SShona.Inf_Out.DEG.DOWN.genes,
  DEG.SShona.Inf.annotation,
  DEG.SShona.Out.annotation,
  indices_to_remove,
  suffixes=c('_SShona_Inf', '_SShona_Out'))

Duke.Inf_Out.DEG.DOWN.exp <- merge_with_annotations(
  Duke.Inf_Out.DEG.DOWN.genes,
  DEG.Duke.Inf.annotation,
  DEG.Duke.Out.annotation,
  indices_to_remove,
  suffixes=c('_Duke_Inf', '_Duke_Out'))
```

## DEG from the comparison of each TOL vs Duke, in infected or outinfected area:

```{r}
# 1 -> gene
# 10 -> Gene_Name
# 8 -> Athaliana_geneId

# 1 -> gene
# 3 -> log2FoldChange
# 7 -> padj

indices_to_remove_from_tb <- c(1, 3, 7)

if (use_v4) {
  indices_to_remove_from_annotation <- c(1, 10, 8)
} else {
  indices_to_remove_from_annotation <- c(2, 11, 10)
}

TOL_vs_Duke.genes <- unique(union(union(DEG.SMira_vs_Duke.Inf$gene, DEG.SMira_vs_Duke.Out$gene),
                                  union(DEG.SShona_vs_Duke.Inf$gene,DEG.SShona_vs_Duke.Out$gene)))

TOL_vs_Duke.annotation <- unique(rbind(rbind(rbind(
                             subset(unique(res.SMira_vs_Duke.Inf.annotation[,indices_to_remove_from_annotation]), gene %in% TOL_vs_Duke.genes),
                             subset(unique(res.SMira_vs_Duke.Out.annotation[,indices_to_remove_from_annotation]), gene %in% TOL_vs_Duke.genes)),
                             subset(unique(res.SShona_vs_Duke.Inf.annotation[,indices_to_remove_from_annotation]), gene %in% TOL_vs_Duke.genes)),
                             subset(unique(res.SShona_vs_Duke.Out.annotation[,indices_to_remove_from_annotation]), gene %in% TOL_vs_Duke.genes)))

TOL_vs_Duke.Inf.Out <- merge(merge(merge(merge(TOL_vs_Duke.annotation,
                             subset(as.data.frame(res.SMira_vs_Duke.Inf.tb[,indices_to_remove_from_tb]), gene %in% TOL_vs_Duke.genes),by = "gene", all.x =T),
                             subset(as.data.frame(res.SMira_vs_Duke.Out.tb[,indices_to_remove_from_tb]), gene %in% TOL_vs_Duke.genes),by = "gene", all.x =T),
                             subset(as.data.frame(res.SShona_vs_Duke.Inf.tb[,indices_to_remove_from_tb]), gene %in% TOL_vs_Duke.genes),by = "gene", all.x =T),
                             subset(as.data.frame(res.SShona_vs_Duke.Out.tb[,indices_to_remove_from_tb]), gene %in% TOL_vs_Duke.genes),by = "gene", all.x =T)

colnames(TOL_vs_Duke.Inf.Out) <- c("gene","name", ifelse(use_v4, "Athaliana_geneID", "TAIR"),
                                   "SMira.Inf.LFC","SMira.Inf.padj","SMira.Out.LFC","SMira.Out.padj",
                                   "SShona.Inf.LFC","SShona.Inf.padj","SShona.Out.LFC","SShona.Out.padj")
```

## Plotting the normalized gene counts across the different time points.

It is a graph for an individual gene.

```{r, echo=FALSE, fig.height=5, fig.width=20}
plotCounts(dds_Blight, gene= "Soltu.DM.01G027350", intgroup=c("variety","treatment"))
```

## Selection of the top 20:

NOTE: temporarily disabled - need to rework the indices for v6 use!

Sarpo Mira (Inf/Out vs Mock):

```{r, echo=FALSE, fig.height=5, fig.width=20}
# DEG.SMira.Inf.annotation.up <- unique(subset(DEG.SMira.Inf.annotation,log2FoldChange > 1)[,c(1,3,7,8,10)])
# DEG.SMira.Out.annotation.up <- unique(subset(DEG.SMira.Out.annotation,log2FoldChange > 1)[,c(1,3,7,8,10)])
# 
# DEG.SMira.Inf.annotation.down <- unique(subset(DEG.SMira.Inf.annotation,log2FoldChange < -1)[,c(1,3,7,8,10)])
# DEG.SMira.Out.annotation.down <- unique(subset(DEG.SMira.Out.annotation,log2FoldChange < -1)[,c(1,3,7,8,10)])
```

Duke (Inf/Out vs Mock):

```{r, echo=FALSE, fig.height=5, fig.width=20}
# DEG.Duke.Inf.annotation.up <- unique(subset(DEG.Duke.Inf.annotation,log2FoldChange > 1)[,c(1,3,7,8,10)])
# DEG.Duke.Out.annotation.up <- unique(subset(DEG.Duke.Out.annotation,log2FoldChange > 1)[,c(1,3,7,8,10)])
# 
# DEG.Duke.Inf.annotation.down <- unique(subset(DEG.Duke.Inf.annotation,log2FoldChange < -1)[,c(1,3,7,8,10)])
# DEG.Duke.Out.annotation.down <- unique(subset(DEG.Duke.Out.annotation,log2FoldChange < -1)[,c(1,3,7,8,10)])
```

Sarpo Mira vs Duke, Inf48

```{r}
# DEG.SMira_vs_Duke.Inf.annotation.up <- unique(subset(DEG.SMira_Duke.Inf.annotation, log2FoldChange > 1)[,c(1,3,7,8,10)])
# DEG.SMira_vs_Duke.Inf.annotation.down <- unique(subset(DEG.SMira_Duke.Inf.annotation, log2FoldChange < -1)[,c(1,3,7,8,10)])
```

Sarpo Mira vs Duke, OutInf48h

```{r}
# DEG.SMira_vs_Duke.Out.annotation.up <- unique(subset(DEG.SMira_Duke.Out.annotation, log2FoldChange > 1)[,c(1,3,7,8,10)])
# DEG.SMira_vs_Duke.Out.annotation.down <- unique(subset(DEG.SMira_Duke.Out.annotation, log2FoldChange < -1)[,c(1,3,7,8,10)])
```

Sarpo Shona vs Duke, Inf48h

```{r}
# DEG.SShona_vs_Duke.Inf.annotation.up <- unique(subset(DEG.SShona_Duke.Inf.annotation, log2FoldChange > 1)[,c(1,3,7,8,10)])
# DEG.SShona_vs_Duke.Inf.annotation.down <- unique(subset(DEG.SShona_Duke.Inf.annotation, log2FoldChange < -1)[,c(1,3,7,8,10)])
```

Sarpo Mira vs Duke, OutInf48h

```{r}
# DEG.SShona_vs_Duke.Out.annotation.up <- unique(subset(DEG.SShona_Duke.Out.annotation, log2FoldChange > 1)[,c(1,3,7,8,10)])
# DEG.SShona_vs_Duke.Out.annotation.down <- unique(subset(DEG.SShona_Duke.Out.annotation, log2FoldChange < -1)[,c(1,3,7,8,10)])
```

## Saving data

```{r}
#For gProfiler2, without annotation:

#save(DEG.Duke.Inf.UP, DEG.Duke.Inf.DOWN, DEG.Duke.Out.UP, DEG.Duke.Out.DOWN,
#     DEG.SMira.Inf.UP, DEG.SMira.Inf.DOWN, DEG.SMira.Out.UP, DEG.SMira.Out.DOWN,
#     DEG.SShona.Inf.UP, DEG.SShona.Inf.DOWN, DEG.SShona.Out.UP, DEG.SShona.Out.DOWN, file = "2.DESeq2_output_48h/DEG_48h_vs_Mock_UpDown.RData")


#save(DEG_p.Duke.Inf.UP, DEG_p.Duke.Inf.DOWN, DEG_p.Duke.Out.UP, DEG_p.Duke.Out.DOWN,
#     DEG_p.SMira.Inf.UP, DEG_p.SMira.Inf.DOWN, DEG_p.SMira.Out.UP, DEG_p.SMira.Out.DOWN,
#     DEG_p.SShona.Inf.UP, DEG_p.SShona.Inf.DOWN, DEG_p.SShona.Out.UP, DEG_p.SShona.Out.DOWN, file = "2.DESeq2_output_48h/DEG_p_48h_vs_Mock_UpDown.RData")





#DEG with their annotation(gene_name, A.thaliana_ID, GOterm) to later select DEGs in base of their linked GO terms:
#save(DEG.Duke.Inf.annotation,DEG.Duke.Out.annotation,
#     DEG.SMira.Inf.annotation,DEG.SMira.Out.annotation,
#     DEG.SShona.Inf.annotation,DEG.SShona.Out.annotation, file ="2.DESeq2_output_48h/DEG_48h_vs_Mock.Annotation_.RData")

#DEG with without GOterms to observe only the gene_name and the A.thaliana_ID. 
#It generate a smaller tables than the previous one to observe better which genes are DEGs:
#write.csv(unique(DEG.Duke.Inf.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_1Duke_Inf48h_vs_Mock_GeneFuntion_notGO.csv")
#write.csv(unique(DEG.Duke.Out.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_1Duke_Out48h_vs_Mock_GeneFuntion_notGO.csv")

#write.csv(unique(DEG.SMira.Inf.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_1SMira_Inf48h_vs_Mock_GeneFuntion_notGO.csv")
#write.csv(unique(DEG.SMira.Out.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_1SMira_Out48h_vs_Mock_GeneFuntion_notGO.csv")

#write.csv(unique(DEG.SShona.Inf.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_1SShona_Inf48h_vs_Mock_GeneFuntion_notGO.csv")
#write.csv(unique(DEG.SShona.Out.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_1SShona_Out48h_vs_Mock_GeneFuntion_notGO.csv")


#To complete the expression of the genes in the tables that will be generated:
#save(res.Duke.Inf, res.Duke.Out, 
#     res.SMira.Inf, res.SMira.Out, 
#     res.SShona.Inf, res.SShona.Out, file = "2.DESeq2_output_48h/res.48h_vs_Mock.RData")

#save(res.Duke.Inf.annotation, res.Duke.Out.annotation, 
#     res.SMira.Inf.annotation, res.SMira.Out.annotation, 
#     res.SShona.Inf.annotation, res.SShona.Out.annotation, file = "2.DESeq2_output_48h/res.48h_vs_Mock.annotation.RData")


#Saving the DEGs from the comparison between each TOL vs Duke in infected or outinfected area:
#write.csv(TOL_vs_Duke.Inf.Out, file = "EachDEG.TOL_vs_Duke_Inf_Out.csv")


#DEG_p:
#save(DEG_p.Duke.Inf.annotation,DEG_p.Duke.Out.annotation,
#     DEG_p.SMira.Inf.annotation,DEG_p.SMira.Out.annotation,
#     DEG_p.SShona.Inf.annotation,DEG_p.SShona.Out.annotation, file ="2.DESeq2_output_48h/DEG_p_Annotation_48h_vs_Mock.RData")

#write.csv(unique(DEG_p.Duke.Inf.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_p_Duke_Inf48h_vs_Mock_GeneFuntion_notGO.csv")
#write.csv(unique(DEG_p.Duke.Out.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_p_Duke_Out48h_vs_Mock_GeneFuntion_notGO.csv")

#write.csv(unique(DEG_p.SMira.Inf.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_p_SMira_Inf48h_vs_Mock_GeneFuntion_notGO.csv")
#write.csv(unique(DEG_p.SMira.Out.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_p_SMira_Out48h_vs_Mock_GeneFuntion_notGO.csv")

#write.csv(unique(DEG_p.SShona.Inf.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_p_SShona_Inf48h_vs_Mock_GeneFuntion_notGO.csv")
#write.csv(unique(DEG_p.SShona.Out.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_p_SShona_Out48h_vs_Mock_GeneFuntion_notGO.csv")

#Saving normalized counts: (from "Norm.Counts <- counts(dds_Blight, normalized=TRUE)")
#write.csv(Norm.Counts, file= "NormalizedCounts.csv")

```
