---
title: "Transcriptomic analysis of potato samples infected with P. infestans - DE analysis"
author: "Patricia Ponce"
date: "June, 2021"
output: 
  html_document:
    self_contained: false
fontsize: 15pt
editor_options: 
  markdown: 
    wrap: 72
---

The data come from potato leaves infected with Phythophtora infestans
for 48 horas using a detached infection assay. Before this analysis the
reads were mapped to the potato genome v6 with STAR, then counting with
HTSeq. Three potato varieties were infected: Sarpo Shona(Tolerant),
Sarpo Mira(Tolerant) and Duke of York(Susceptible).

PURPOSE: This part of the transcriptomic analysis involve the
normalization of the counts by DESeq2. After that we will plot a PCA to
visualize the distribution of the samples, then find which genes are
significant different between varieties during 48hours after infection.
All the comparison will be done with against the mock (Infected-Area vs
Mock or Out-infection-Area vs Mock).

INPUT: - HTseq counts in "5.HTSEQ_blight". - Annotation info from the
potato genone v6.1 in
"Z:/1.PotatoDroughtOut_genomev6/4.GO_analysis/gProfile_FileHC.GeneTransID.Name.GO/potv6_Ath_potv404.RData".

OUTPUT:\
-"1.DESeq2_analysis_blight_infection48h.html" (After kniting)
-"1.DESeq2_analysis_blight_infection48h_files" --\> Inside the folder
"figure-html" are all the graphs produced with this code. (After
kniting) -All the files in the "2.DESeq2_output_48h" folder

Path of the folder for the inputs and outputs:

```{r}
setwd("/rds/projects/l/leachlj-potato-qtl-project/M6_Area/potato-blight-rna-seq-pipeline")
```

Packages to employ for the analysis:

```{r message=FALSE, warning=FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("apeglm")
```


```{r message=FALSE, warning=FALSE}
#this code works to install an older version of lasso2 from the archives
packageurl <- "https://cran.r-project.org/src/contrib/Archive/lasso2/lasso2_1.2-22.tar.gz"
install.packages(packageurl, repos=NULL, type="source")
```

```{r message=FALSE, warning=FALSE}
BiocManager::install("DEGreport")
```

```{r}
BiocManager::install("EnhancedVolcano")
BiocManager::install("ComplexHeatmap")
```

```{r}
library(apeglm)
library(DESeq2)
library(DEGreport)
library(EnhancedVolcano)
library(ggplot2)
library(pheatmap)
library(RColorBrewer)
library(pheatmap)
library(dplyr) 
library(tibble)
library(data.table)
library(gridExtra)
library(stringr)
library(VennDiagram)
library(ComplexHeatmap)
library(InteractiveComplexHeatmap)
library(biomaRt)


library(devtools)
install_github("jokergoo/InteractiveComplexHeatmap")

if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("InteractiveComplexHeatmap")

library(InteractiveComplexHeatmap)

BiocManager::install("biomaRt")

```

```{r}
#specify this to enable plotting to work
options(bitmapType='cairo')
```

Preparing a table with the input files from HTSeq to run DESeq2 :

```{r}
directory = './quantified'
sampleFiles = grep('*count.txt', list.files(directory), value = TRUE)
sampleFilesR = sort(sampleFiles[startsWith(sampleFiles, 'R')])
sampleFilesA = sort(sampleFiles[startsWith(sampleFiles, 'A')])
sampleFiles = c(sampleFilesR, sampleFilesA)
sampleNames = c(
  'SMira_C1', 'SMira_C2', 'SMira_C3',
  'SMira_OI1', 'SMira_OI2', 'SMira_OI3',
  'Duke_C1', 'Duke_C2', 'Duke_C3',
  'Duke_OI1', 'Duke_OI2', 'Duke_OI3',
  'SShona_C1', 'SShona_C2', 'SShona_C3',
  'SShona_OI1', 'SShona_OI2', 'SShona_OI3',
  'SMira_I1', 'SMira_I2', 'SMira_I3',
  'Duke_I1', 'Duke_I2', 'Duke_I3',
  'SShona_I1', 'SShona_I2', 'SShona_I3',
  'SMira_M1', 'SMira_M2', 'SMira_M3',
  'Duke_M1', 'Duke_M2', 'Duke_M3',
  'SShona_M1', 'SShona_M2', 'SShona_M3'
)

sampleReplicate = rep(c(1, 2, 3), 12)

sampleVariety = c(
  rep('SMira', 6),
  rep('Duke', 6),
  rep('SShona', 6),
  rep('SMira', 3),
  rep('Duke', 3),
  rep('SShona', 3),
  rep('SMira', 3),
  rep('Duke', 3),
  rep('SShona', 3)
)

sampleTreatment = c(
  rep('Control', 3),
  rep('OutInf', 3),
  rep('Control', 3),
  rep('OutInf', 3),
  rep('Control', 3),
  rep('OutInf', 3),
  rep('Inf', 9),
  rep('Mock', 9)
)

sampleTable = data.frame(sampleNames= sampleNames, 
                          fileName = sampleFiles, 
                          replicate = sampleReplicate,
                          variety = sampleVariety,
                          treatment = sampleTreatment)
sampleTable
```

```{r}
sampleTable$replicate = factor(sampleTable$replicate)
sampleTable$variety = factor(sampleTable$variety)
sampleTable$treatment = factor(sampleTable$treatment)
```

## Running DESeq2 with all the samples.

```{r}
dds_Blight <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable, 
                                       directory = directory,
                                       design = ~variety + treatment + variety:treatment)

# Removing genes whose counts is less than 10 in total around all the libraries
keep <- rowSums(counts(dds_Blight)) >= 10
dds_Blight <- dds_Blight[keep,]

dds_Blight$variety <- relevel(dds_Blight$variety, ref = c("Duke"))
dds_Blight$treatment <- relevel(dds_Blight$treatment, ref = c("Mock"))
#dds_Blight$treatment <- relevel(dds_Blight$treatment, ref = c("Control"))

dds_Blight <- DESeq(dds_Blight)
resultsNames(dds_Blight) #it's the input for "name" and "constrast"
```

## Normalized counts

```{r}
Norm.Counts <- counts(dds_Blight, normalized=TRUE)
```

## Gene count distribution per samples in leaf.

Distribution before normalization:

```{r}
boxplot(log2(counts(dds_Blight)+1), range=0, las=2)
```

Distribution after normalization:

```{r}
boxplot(log2(counts(dds_Blight, normalized=TRUE)+1), range=0, las=2)
```

## Principal component analysis (PCA) of the leaf samples.

Variance stabilization transformation:

```{r}
vst_Blight <- vst(dds_Blight, blind=TRUE)
```

PCA of all leaf samples:

```{r, echo=FALSE, fig.height=7, fig.width=14}

# n = 29393
pcaVST <- plotPCA(vst_Blight, intgroup = c("treatment","variety"), returnData = TRUE, ntop = 29393)
percentVar <- round(100*attr(pcaVST, "percentVar"))

SusTol_all <- qplot(PC1, PC2, color=treatment, shape=variety, data=pcaVST) +
  geom_point(size=3)+
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))+
  labs(title="All leaf samples_n29393")+
  geom_point()+
  geom_text(aes(label=colnames(vst_Blight)),size = 3, vjust=1.5, hjust=1)


# n = 1000
pcaVST_n1000 <- plotPCA(vst_Blight, intgroup = c("treatment","variety"), returnData = TRUE, ntop = 1000)
percentVar_n1000 <- round(100*attr(pcaVST_n1000, "percentVar"))

SusTol_n1000 <- qplot(PC1, PC2, color=treatment, shape=variety, data=pcaVST_n1000) +
           geom_point(size=3)+
           xlab(paste0("PC1: ",percentVar_n1000[1],"% variance")) +
           ylab(paste0("PC2: ",percentVar_n1000[2],"% variance"))+
           labs(title="All leaf samples_n1000")+
           geom_point()+
           geom_text(aes(label=colnames(vst_Blight)),size = 3, vjust=1.5, hjust=1)

grid.arrange(SusTol_all, SusTol_n1000,
             ncol=2, widths=unit(c(160,160), c("mm","mm")), heights=unit(150, "mm"))
```

The distribution of the samples is very similar in both cases, by using
all the normalized genes or using only the 1000 most variable genes. All
replicates looks good, then it won't be necessary to remove any sample.

## Heatmap and clustering of the samples:

Heatmap of all leaf samples, after removing bad biological replicates:

```{r}
sampleDists <- dist(t(assay(vst_Blight)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vst_Blight$variety, vst_Blight$treatment, sep="-")
colnames(sampleDistMatrix) <- NULL

colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)

Blighthm <- pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)
print(Blighthm)


sampleDists <- dist(t(assay(vst_Blight)))
sampleDistM <- as.matrix((sampleDists))

Variety_comparisons <- pheatmap(sampleDistM)

topVarGenes <- head( order( rowVars( assay(vst_Blight) ), decreasing=TRUE ), 100 )

Variety_genes <- ComplexHeatmap::heatmap.2( assay(vst_Blight)[ topVarGenes, ], cexRow = 0.3, cexCol = 0.7, margins = c(1,1), main = "Variety and Treatment vs Top 100 Genes",
                      trace="none",
col = colorRampPalette( rev(brewer.pal(9, "RdBu")) )(255))  




blight_tidy <- Blight |> as_tibble(rownames="Genes")

#vst_Blight%>%colnames
#columns are the conditions
#rows are genes
  
)


```

```{r}
assignInNamespace("pheatmap", ComplexHeatmap::pheatmap, ns = "pheatmap")


ComplexHeatmap::pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)


ht1 <- ComplexHeatmap::pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)
ht1 = draw(ht1)

ui = fluidPage(
    h3("My first interactive ComplexHeatmap Shiny app"),
    p("This is an interactive heatmap visualization on a random matrix."),
    InteractiveComplexHeatmapOutput()
)

server = function(input, output, session) {
    makeInteractiveComplexHeatmap(input, output, session, ht1)
}

shinyApp(ui, server)

ht_shiny(ht1)

```
## Pairwise comparisons:

#Tutorial:
<https://rstudio-pubs-static.s3.amazonaws.com/329027_593046fb6d7a427da6b2c538caf601e1.html>

Data to compare:

```{r}
resultsNames(dds_Blight) #it's the input for "name" and "contrast"
```

For the comparison: - Base_level: Duke / Mock - res: To get the pvalue
and lfcThreshold=0, - default lfcThreshold=0

For the DE analysis, genes with the absolute value of the log2folchange
more than 1 (\|log2FC\| \> 1) will be selected.

Pairwise comparisons between samples (Inf/OutInf vs Mock) from the
tolerant variety:

```{r}
res.Duke.Out  <- results(dds_Blight, alpha = 0.05, contrast = list(c("treatment_OutInf_vs_Mock"))) 
res.Duke.Inf  <- results(dds_Blight, alpha = 0.05, contrast = list(c("treatment_Inf_vs_Mock"))) 
res.SMira.Out <- results(dds_Blight, alpha = 0.05, contrast = list(c("treatment_OutInf_vs_Mock","varietySMira.treatmentOutInf"))) 
res.SMira.Inf <- results(dds_Blight, alpha = 0.05, contrast = list(c("treatment_Inf_vs_Mock","varietySMira.treatmentInf"))) 
res.SShona.Out<- results(dds_Blight, alpha = 0.05, contrast = list(c("treatment_OutInf_vs_Mock","varietySShona.treatmentOutInf"))) 
res.SShona.Inf<- results(dds_Blight, alpha = 0.05, contrast = list(c("treatment_Inf_vs_Mock","varietySShona.treatmentInf"))) 
```

Pairwise comparisons between samples (control vs mock) from all varieties:


Pairwise comparisons between samples (tolerant vs Duke in Infect and
OutInfect. area) from the tolerant variety:

```{r}
res.SMira_vs_Duke.Out <- results(dds_Blight, alpha = 0.05, contrast = list(c("variety_SMira_vs_Duke" ,"varietySMira.treatmentOutInf"))) 
res.SMira_vs_Duke.Inf <- results(dds_Blight, alpha = 0.05, contrast = list(c("variety_SMira_vs_Duke" ,"varietySMira.treatmentInf"))) 
res.SShona_vs_Duke.Out<- results(dds_Blight, alpha = 0.05, contrast = list(c("variety_SShona_vs_Duke","varietySShona.treatmentOutInf"))) 
res.SShona_vs_Duke.Inf<- results(dds_Blight, alpha = 0.05, contrast = list(c("variety_SShona_vs_Duke","varietySShona.treatmentInf"))) 
```

## Volcano plots:

This plot allows as to observe the ranges of the foldchanges and
visualize if a high or low amount of genes pass the threshold of
\|log2FC\| \> 1 & padj \< 0.05. In this case, the foldchange was
calculated from the OutINf/Mock and INf/Mock in each variety.

In all the samples:

```{r, echo=FALSE, fig.height=20, fig.width=14}
volcano.Duke.Inf <- EnhancedVolcano(res.Duke.Inf, 
                lab = rownames(res.Duke.Inf),
                x = 'log2FoldChange',
                y = 'padj',
                pCutoff = 0.05,
                FCcutoff = 1,
                xlim = c(-5, 8), title = "res.Duke.Inf")

volcano.Duke.Out <- EnhancedVolcano(res.Duke.Out, 
                lab = rownames(res.Duke.Out),
                x = 'log2FoldChange',
                y = 'padj',
                pCutoff = 0.05,
                FCcutoff = 1,
                xlim = c(-5, 8), title = "res.Duke.Out")

volcano.SMira.Inf <- EnhancedVolcano(res.SMira.Inf, 
                lab = rownames(res.SMira.Inf),
                x = 'log2FoldChange',
                y = 'padj',
                pCutoff = 0.05,
                FCcutoff = 1,
                xlim = c(-5, 8), title = "res.SMira.Inf")

volcano.SMira.Out <- EnhancedVolcano(res.SMira.Out, 
                lab = rownames(res.SMira.Out),
                x = 'log2FoldChange',
                y = 'padj',
                pCutoff = 0.05,
                FCcutoff = 1,
                xlim = c(-5, 8), title = "res.SMira.Out")

volcano.SShona.Inf <- EnhancedVolcano(res.SShona.Inf, 
                lab = rownames(res.SShona.Inf),
                x = 'log2FoldChange',
                y = 'padj',
                pCutoff = 0.05,
                FCcutoff = 1,
                xlim = c(-5, 8), title = "res.SShona.Inf")

volcano.SShona.Out <- EnhancedVolcano(res.SShona.Out, 
                lab = rownames(res.SShona.Out),
                x = 'log2FoldChange',
                y = 'padj',
                pCutoff = 0.05,
                FCcutoff = 1,
                xlim = c(-5, 8), title = "res.SShona.Out")

grid.arrange(volcano.Duke.Inf, volcano.Duke.Out,
             volcano.SMira.Inf, volcano.SMira.Out,
             volcano.SShona.Inf, volcano.SShona.Out,
             ncol = 2, nrow = 3, widths =unit(c(160,160), c("mm","mm")), heights = unit(c(150,150,150), c("mm","mm","mm")))
```

Control vs Mock

```{r, echo=FALSE, fig.height=20, fig.width=14}

res.Duke.CvM  <- results(dds_Blight, alpha = 0.05, contrast = list(c("treatment_Control_vs_Mock"))) 

res.SMira.CvM <- results(dds_Blight, alpha = 0.05, contrast = list(c("treatment_Control_vs_Mock","varietySMira.treatmentControl"))) 

res.SShona.CvM <- results(dds_Blight, alpha = 0.05, contrast = list(c("treatment_Control_vs_Mock","varietySShona.treatmentControl"))) 


res.Duke.CvM.vp <- EnhancedVolcano(res.Duke.CvM, 
                lab = rownames(res.Duke.CvM),
                x = 'log2FoldChange',
                y = 'padj',
                pCutoff = 0.05,
                FCcutoff = 1,
                xlim = c(-5, 8), title = "res.Duke.CvM")

res.SMira.CvM.vp <- EnhancedVolcano(res.SMira.CvM, 
                lab = rownames(res.SMira.CvM),
                x = 'log2FoldChange',
                y = 'padj',
                pCutoff = 0.05,
                FCcutoff = 1,
                xlim = c(-5, 8), title = "res.SMira.CvM")

res.SShona.CvM.vp <- EnhancedVolcano(res.SShona.CvM, 
                lab = rownames(res.SShona.CvM),
                x = 'log2FoldChange',
                y = 'padj',
                pCutoff = 0.05,
                FCcutoff = 1,
                xlim = c(-5, 8), title = "res.SShona.CvM")

grid.arrange(res.Duke.CvM.vp,res.SMira.CvM.vp, res.SShona.CvM.vp,
             ncol = 3, nrow = 1, widths =unit(c(160,160,160), c("mm","mm","mm")), heights = unit(c(450), c("mm")))

#Functions for DEG

to_tibble <- function (.data) {
  data.frame(.data) %>%
    rownames_to_column(var="gene") %>%
    as_tibble()
}

select_degs_only_padj <- function(.data) {
  to_tibble(.data) %>%
    filter(padj < 0.05)
}

select_degs <- function(.data) {
  to_tibble(.data) %>%
    filter(padj < 0.05) %>%
    filter(abs(log2FoldChange) > 1)
}

#DEG selection

DEG.Duke.CvM <- res.Duke.CvM %>% select_degs()
DEG_p.Duke.CvM <- res.Duke.CvM %>% select_degs_only_padj()

DEG.SMira.CvM <- res.SMira.CvM %>% select_degs()
DEG_p.SMira.CvM <- res.SMira.CvM %>% select_degs_only_padj()

DEG.SShona.CvM <- res.SShona.CvM %>% select_degs()
DEG_p.SShona.CvM <- res.SShona.CvM %>% select_degs_only_padj()


#Venn Diagrams

DEG.Duke.CvM.UP     <- subset(DEG.Duke.CvM, log2FoldChange > 1)
DEG_p.Duke.CvM.UP   <- subset(DEG_p.Duke.CvM, log2FoldChange > 1)
DEG.Duke.CvM.DOWN   <- subset(DEG.Duke.CvM, log2FoldChange < -1)
DEG_p.Duke.CvM.DOWN <- subset(DEG_p.Duke.CvM, log2FoldChange < -1)

DEG.SMira.CvM.UP    <- subset(DEG.SMira.CvM, log2FoldChange > 1)
DEG_p.SMira.CvM.UP  <- subset(DEG_p.SMira.CvM, log2FoldChange > 1)
DEG.SMira.CvM.DOWN  <- subset(DEG.SMira.CvM, log2FoldChange < -1)
DEG_p.SMira.CvM.DOWN<- subset(DEG_p.SMira.CvM, log2FoldChange < -1)

DEG.SShona.CvM.UP   <- subset(DEG.SShona.CvM, log2FoldChange > 1)
DEG_p.SShona.CvM.UP <- subset(DEG_p.SShona.CvM, log2FoldChange > 1)
DEG.SShona.CvM.DOWN <- subset(DEG.SShona.CvM, log2FoldChange < -1)
DEG_p.SShona.CvM.DOWN <- subset(DEG_p.SShona.CvM, log2FoldChange < -1)





Venn.CvM.DEG.UP <- venn.diagram(filename = NULL, 
             list(Duke_CvM_Up   = DEG.Duke.CvM.UP$gene, 
                  S.Mira_CvM_Up = DEG.SMira.CvM.UP$gene,
                  S.Shona_CvM_Up = DEG.SShona.CvM.UP$gene), 
             scaled = FALSE, col = "black", fill = c("#E69F00","#56B4E9", "#009E73"),main = "Control vs Mock - Up",
             euler.d = FALSE,lty = 1 , cex = 0.9, lwd = 1, cat.cex = 1, sigdigs = 1,
             print.mode=c("raw","percent"), height = 20, width = 20, cat.pos = c(-20,20,-180), cat.dist = c(0.06, 0.06, 0.04))

Venn.CvM.DEG.DOWN <- venn.diagram(filename = NULL, 
             list(Duke_CvM_Down   = DEG.Duke.CvM.DOWN$gene, 
                  S.Mira_CvM_Down = DEG.SMira.CvM.DOWN$gene,
                  S.Shona_CvM_Down = DEG.SShona.CvM.DOWN$gene), 
             scaled = FALSE, col = "black", fill = c("#E69F00","#56B4E9", "#009E73"),main = "Control vs Mock - Down",
             euler.d = FALSE,lty = 1 , cex = 0.9, lwd = 1, cat.cex = 1, sigdigs = 1,
             print.mode=c("raw","percent"), height = 20, width = 20, cat.pos = c(-20,20,-180), cat.dist = c(0.06, 0.06, 0.04))

grid.arrange(gTree(children = Venn.CvM.DEG.UP),  gTree(children = Venn.CvM.DEG.DOWN), 
             top = "Control vs Mock Differential Expression by Variety", ncol = 2, nrow = 1, 
             widths =unit(c(200,200), c("mm","mm")), 
             heights = unit(c(200), c("mm")))

#GO Annotations and Enrichment

DEG.Duke.CvM.annotation <- go_and_func_enrichment(DEG.Duke.CvM)
DEG_p.Duke.CvM.annotation <- go_and_func_enrichment(DEG_p.Duke.CvM)

DEG.SMira.CvM.annotation <- go_and_func_enrichment(DEG.SMira.CvM)
DEG_p.SMira.CvM.annotation <- go_and_func_enrichment(DEG_p.SMira.CvM)

DEG.SShona.CvM.annotation <- go_and_func_enrichment(DEG.SShona.CvM)
DEG_p.SShona.CvM.annotation <- go_and_func_enrichment(DEG_p.SShona.CvM)




```

## Selecting differentially expressed genes (DEG):

First we define some useful filtering/transformation functions:

```{r}
to_tibble <- function (.data) {
  data.frame(.data) %>%
    rownames_to_column(var="gene") %>%
    as_tibble()
}

select_degs_only_padj <- function(.data) {
  to_tibble(.data) %>%
    filter(padj < 0.05)
}

select_degs <- function(.data) {
  to_tibble(.data) %>%
    filter(padj < 0.05) %>%
    filter(abs(log2FoldChange) > 1)
}
```

(1)Inf/OutInf vs Mock in each variety. Selection: padj. \< 0.05 & log2FC
\> 1:

```{r}
DEG.Duke.Out   <- res.Duke.Out %>% select_degs()
DEG_p.Duke.Out <- res.Duke.Out %>% select_degs_only_padj()

DEG.Duke.Inf   <- res.Duke.Inf %>% select_degs()
DEG_p.Duke.Inf <- res.Duke.Inf %>% select_degs_only_padj()

DEG.SMira.Out  <- res.SMira.Out %>% select_degs()
DEG_p.SMira.Out <- res.SMira.Out %>% select_degs_only_padj()

DEG.SMira.Inf  <- res.SMira.Inf %>% select_degs()
DEG_p.SMira.Inf <- res.SMira.Inf %>% select_degs_only_padj()

DEG.SShona.Out <- res.SShona.Out %>% select_degs()
DEG_p.SShona.Out <- res.SShona.Out %>% select_degs_only_padj()

DEG.SShona.Inf <- res.SShona.Inf %>% select_degs()
DEG_p.SShona.Inf <- res.SShona.Inf %>% select_degs_only_padj()
```

(2)Tol vs Duke in infected or outinfected area. padj. \< 0.05 & log2FC
\> 1:

```{r}
DEG.SMira_vs_Duke.Out  <- res.SMira_vs_Duke.Out %>% select_degs()
DEG.SMira_vs_Duke.Inf  <- res.SMira_vs_Duke.Inf %>% select_degs()
DEG.SShona_vs_Duke.Out <- res.SShona_vs_Duke.Out %>% select_degs()
DEG.SShona_vs_Duke.Inf <- res.SShona_vs_Duke.Inf %>% select_degs()
```

## Venn diagrams

Getting gene names of up and downregulated genes:

```{r}
#Up and downregulated genes with a padj < 0.05 & |LFC| > 1: DEG...............................................................................................................
DEG.Duke.Inf.UP     <- subset(DEG.Duke.Inf, log2FoldChange > 1)
DEG_p.Duke.Inf.UP   <- subset(DEG_p.Duke.Inf, log2FoldChange > 1)
DEG.Duke.Inf.DOWN   <- subset(DEG.Duke.Inf, log2FoldChange < -1)
DEG_p.Duke.Inf.DOWN <- subset(DEG_p.Duke.Inf, log2FoldChange < -1)

DEG.Duke.Out.UP     <- subset(DEG.Duke.Out, log2FoldChange > 1)
DEG_p.Duke.Out.UP   <- subset(DEG_p.Duke.Out, log2FoldChange > 1)
DEG.Duke.Out.DOWN   <- subset(DEG.Duke.Out, log2FoldChange < -1)
DEG_p.Duke.Out.DOWN <- subset(DEG_p.Duke.Out, log2FoldChange < -1)

DEG.SMira.Inf.UP    <- subset(DEG.SMira.Inf, log2FoldChange > 1)
DEG_p.SMira.Inf.UP  <- subset(DEG_p.SMira.Inf, log2FoldChange > 1)
DEG.SMira.Inf.DOWN  <- subset(DEG.SMira.Inf, log2FoldChange < -1)
DEG_p.SMira.Inf.DOWN<- subset(DEG_p.SMira.Inf, log2FoldChange < -1)

DEG.SMira.Out.UP    <- subset(DEG.SMira.Out, log2FoldChange > 1)
DEG_p.SMira.Out.UP  <- subset(DEG_p.SMira.Out, log2FoldChange > 1)
DEG.SMira.Out.DOWN  <- subset(DEG.SMira.Out, log2FoldChange < -1)
DEG_p.SMira.Out.DOWN<- subset(DEG_p.SMira.Out, log2FoldChange < -1)

DEG.SShona.Inf.UP   <- subset(DEG.SShona.Inf, log2FoldChange > 1)
DEG_p.SShona.Inf.UP <- subset(DEG_p.SShona.Inf, log2FoldChange > 1)
DEG.SShona.Inf.DOWN <- subset(DEG.SShona.Inf, log2FoldChange < -1)
DEG_p.SShona.Inf.DOWN <- subset(DEG_p.SShona.Inf, log2FoldChange < -1)

DEG.SShona.Out.UP   <- subset(DEG.SShona.Out, log2FoldChange > 1)
DEG_p.SShona.Out.UP   <- subset(DEG_p.SShona.Out, log2FoldChange > 1)
DEG.SShona.Out.DOWN <- subset(DEG.SShona.Out, log2FoldChange < -1)
DEG_p.SShona.Out.DOWN <- subset(DEG_p.SShona.Out, log2FoldChange < -1)


#DEGs Up and downregulated from the comparison of each TOL vs Duke in Inf. and Out.Infected.................................................................................

DEG.SMira_vs_Duke.Inf.UP     <- subset(DEG.SMira_vs_Duke.Inf, log2FoldChange > 1)
DEG.SMira_vs_Duke.Inf.DOWN   <- subset(DEG.SMira_vs_Duke.Inf, log2FoldChange < -1)

DEG.SShona_vs_Duke.Inf.UP     <- subset(DEG.SShona_vs_Duke.Inf, log2FoldChange > 1)
DEG.SShona_vs_Duke.Inf.DOWN   <- subset(DEG.SShona_vs_Duke.Inf, log2FoldChange < -1)

DEG.SMira_vs_Duke.Out.UP     <- subset(DEG.SMira_vs_Duke.Out, log2FoldChange > 1)
DEG.SMira_vs_Duke.Out.DOWN   <- subset(DEG.SMira_vs_Duke.Out, log2FoldChange < -1)

DEG.SShona_vs_Duke.Out.UP     <- subset(DEG.SShona_vs_Duke.Out, log2FoldChange > 1)
DEG.SShona_vs_Duke.Out.DOWN   <- subset(DEG.SShona_vs_Duke.Out, log2FoldChange < -1)

```

(1) Venn diagrams of DEG in each variety, Outinfection and Infection:

```{r, echo=FALSE, fig.height=8, fig.width=13}
#DEGs: |FC|>1, padj. <0.05:...........................................................................................................................................
Venn.Duke.DEG.UP <- venn.diagram(filename = NULL, 
             list(Duke_Out= DEG.Duke.Out.UP$gene, 
                  Duke_Inf= DEG.Duke.Inf.UP$gene), 
             scaled = FALSE, col = "black", fill = c("#EFC000FF","#CD534CFF"), 
             euler.d = FALSE,lty = 1 , cex = 1, lwd = 1, cat.cex = 1, sigdigs = 1,
             print.mode=c("raw","percent"), height = 20, width = 20, main = "Duke of York - Up",
             rotation.degree = 180,
             cat.just=list(c(0.3,-1.5), c(0.7,-1.5)),cat.pos = c(-40,40),cat.dist = c(0.01, 0.01))


Venn.Duke.DEG.DOWN <- venn.diagram(filename = NULL, 
             list(Duke_Out = DEG.Duke.Out.DOWN$gene, 
                  Duke_Inf = DEG.Duke.Inf.DOWN$gene), 
             scaled = FALSE, col = "black", fill = c("#EFC000FF","#CD534CFF"), 
             euler.d = FALSE,lty = 1 , cex = 1, lwd = 1, cat.cex = 1, sigdigs = 1,
             print.mode=c("raw","percent"), height = 20, width = 20, main = "Duke of York - Down",
             rotation.degree = 180,
             cat.just=list(c(0.3,-1.5), c(0.7,-1.5)),cat.pos = c(-40,40),cat.dist = c(0.01, 0.01))

             
Venn.SMira.DEG.UP <- venn.diagram(filename = NULL, 
             list(S.Mira_Out = DEG.SMira.Out.UP$gene, 
                  S.Mira_Inf = DEG.SMira.Inf.UP$gene), 
             scaled = FALSE, col = "black", fill = c("#EFC000FF","#CD534CFF"), 
             euler.d = FALSE,lty = 1 , cex = 1, lwd = 1, cat.cex = 1, sigdigs = 1,
             print.mode=c("raw","percent"), height = 20, width = 20, main = "Sarpo Mira - Up", 
             rotation.degree = 180, 
             cat.just=list(c(0.3,-1.5), c(0.7,-1.5)),cat.pos = c(-40,40),cat.dist = c(0.01, 0.01))

Venn.SMira.DEG.DOWN <- venn.diagram(filename = NULL, 
             list(S.Mira_Out = DEG.SMira.Out.DOWN$gene, 
                  S.Mira_Inf = DEG.SMira.Inf.DOWN$gene), 
             scaled = FALSE, col = "black", fill = c("#EFC000FF","#CD534CFF"), 
             euler.d = FALSE,lty = 1 , cex = 1, lwd = 1, cat.cex = 1, sigdigs = 1,
             print.mode=c("raw","percent"), height = 20, width = 20, main = "Sarpo Mira - Down", 
             rotation.degree = 180,
             cat.just=list(c(0.3,-1.5), c(0.7,-1.5)),cat.pos = c(-40,40),cat.dist = c(0.01, 0.01))


Venn.SShona.DEG.UP <- venn.diagram(filename = NULL, 
             list(S.Shona_Out = DEG.SShona.Out.UP$gene, 
                  S.Shona_Inf = DEG.SShona.Inf.UP$gene), 
             scaled = FALSE, col = "black", fill = c("#EFC000FF","#CD534CFF"), 
             euler.d = FALSE,lty = 1 , cex = 1, lwd = 1, cat.cex = 1, sigdigs = 1,
             rotation.degree = 180,
             print.mode=c("raw","percent"), height = 20, width = 20, main = "Sarpo Shona - Up",
             cat.just=list(c(0.3,-1.5), c(0.7,-1.5)),cat.pos = c(-40,40),cat.dist = c(0.01, 0.01))

Venn.SShona.DEG.DOWN <- venn.diagram(filename = NULL, 
             list(S.Shona_Out = DEG.SShona.Out.DOWN$gene, 
                  S.Shona_Inf = DEG.SShona.Inf.DOWN$gene), 
             scaled = FALSE, col = "black", fill = c("#EFC000FF","#CD534CFF"), 
             euler.d = FALSE,lty = 1 , cex = 1, lwd = 1, cat.cex = 1, sigdigs = 1,
             rotation.degree = 180,
             print.mode=c("raw","percent"), height = 20, width = 20, main = "Sarpo Shona - Down", 
             cat.just=list(c(0.3,-1.5), c(0.7,-1.5)),cat.pos = c(-40,40),cat.dist = c(0.01, 0.01))


#DEGs for only padj. <0.05, not selected by log2FC:...........................................................................................................................
Venn.Duke.DEG_p.UP <- venn.diagram(filename = NULL,
                                 list(Duke_Out= DEG_p.Duke.Out.UP$gene,
                                      Duke_Inf= DEG_p.Duke.Inf.UP$gene),
                                 scaled = FALSE, col = "black", fill = c("#EFC000FF","#CD534CFF"),
                                 euler.d = FALSE,lty = 1 , cex = 1, lwd = 1, cat.cex = 1, sigdigs = 1,
                                 height = 20, width = 20, main = "Duke of York - Up",
                                 cat.just=list(c(0.3,-1.5), c(0.7,-1.5)),cat.pos = c(-40,40),cat.dist = c(0.01, 0.01))


Venn.Duke.DEG_p.DOWN <- venn.diagram(filename = NULL,
                                   list(Duke_Out = DEG_p.Duke.Out.DOWN$gene,
                                        Duke_Inf = DEG_p.Duke.Inf.DOWN$gene),
                                   scaled = FALSE, col = "black", fill = c("#EFC000FF","#CD534CFF"),
                                   euler.d = FALSE,lty = 1 , cex = 1, lwd = 1, cat.cex = 1, sigdigs = 1,
                                   print.mode=c("raw","percent"), height = 20, width = 20, main = "Duke of York - Down",
                                  cat.just=list(c(0.3,-1.5), c(0.7,-1.5)),cat.pos = c(-40,40),cat.dist = c(0.01, 0.01))


Venn.SMira.DEG_p.UP <- venn.diagram(filename = NULL,
                                  list(S.Mira_Out = DEG_p.SMira.Out.UP$gene,
                                       S.Mira_Inf = DEG_p.SMira.Inf.UP$gene),
                                  scaled = FALSE, col = "black", fill = c("#EFC000FF","#CD534CFF"),
                                  euler.d = FALSE,lty = 1 , cex = 1, lwd = 1, cat.cex = 1, sigdigs = 1,
                                  print.mode=c("raw","percent"), height = 20, width = 20, main = "Sarpo Mira - Up",
                                  rotation.DEG_pree = 180, cat.just=list(c(0.3,-1.5), c(0.7,-1.5)),cat.pos = c(-40,40),cat.dist = c(0.01, 0.01))

Venn.SMira.DEG_p.DOWN <- venn.diagram(filename = NULL,
                                    list(S.Mira_Out = DEG_p.SMira.Out.DOWN$gene,
                                         S.Mira_Inf = DEG_p.SMira.Inf.DOWN$gene),
                                    scaled = FALSE, col = "black", fill = c("#EFC000FF","#CD534CFF"),
                                    euler.d = FALSE,lty = 1 , cex = 1, lwd = 1, cat.cex = 1, sigdigs = 1,
                                    print.mode=c("raw","percent"), height = 20, width = 20, main = "Sarpo Mira - Down",
                                    rotation.DEG_pree = 180,cat.just=list(c(0.3,-1.5), c(0.7,-1.5)),cat.pos = c(-40,40),cat.dist = c(0.01, 0.01))


Venn.SShona.DEG_p.UP <- venn.diagram(filename = NULL,
                                   list(S.Shona_Out = DEG_p.SShona.Out.UP$gene,
                                        S.Shona_Inf = DEG_p.SShona.Inf.UP$gene),
                                   scaled = FALSE, col = "black", fill = c("#EFC000FF","#CD534CFF"),
                                   euler.d = FALSE,lty = 1 , cex = 1, lwd = 1, cat.cex = 1, sigdigs = 1,
                                   print.mode=c("raw","percent"), height = 20, width = 20, main = "Sarpo Shona - Up",
                                   rotation.DEG_pree = 180,cat.just=list(c(0.3,-1.5), c(0.7,-1.5)),cat.pos = c(-40,40),cat.dist = c(0.01, 0.01))

Venn.SShona.DEG_p.DOWN <- venn.diagram(filename = NULL,
                                     list(S.Shona_Out = DEG_p.SShona.Out.DOWN$gene,
                                          S.Shona_Inf = DEG_p.SShona.Inf.DOWN$gene),
                                     scaled = FALSE, col = "black", fill = c("#EFC000FF","#CD534CFF"),
                                     euler.d = FALSE,lty = 1 , cex = 1, lwd = 1, cat.cex = 1, sigdigs = 1,
                                     print.mode=c("raw","percent"), height = 20, width = 20, main = "Sarpo Shona - Down",
                                     rotation.DEG_pree = 180,cat.just=list(c(0.3,-1.5), c(0.7,-1.5)),cat.pos = c(-40,40),cat.dist = c(0.01, 0.01))


grid.arrange(gTree(children = Venn.Duke.DEG.UP),  gTree(children = Venn.Duke.DEG.DOWN), 
             gTree(children = Venn.Duke.DEG_p.UP),  gTree(children = Venn.Duke.DEG_p.DOWN),
             gTree(children = Venn.SMira.DEG.UP), gTree(children = Venn.SMira.DEG.DOWN),
             gTree(children = Venn.SMira.DEG_p.UP), gTree(children = Venn.SMira.DEG_p.DOWN),
             gTree(children = Venn.SShona.DEG.UP),gTree(children = Venn.SShona.DEG.DOWN),
             gTree(children = Venn.SShona.DEG_p.UP),gTree(children = Venn.SShona.DEG_p.DOWN),
             top = "Nro. of Genes with a padj.<0.05 & |log2FC| > 1  -  Nro. of Genes with a padj.<0.05", ncol = 4, nrow = 3, 
             widths =unit(c(58,58,58,58), c("mm","mm","mm","mm")), 
             heights = unit(c(58,58,58), c("mm","mm","mm")))
```

(2) Venn diagrams of DEG in each condition:

```{r, echo=FALSE, fig.height=8, fig.width=13}
#DEGs (|FC|>1, padj. <0.05)
Venn.OutInf.DEG.UP <- venn.diagram(filename = NULL, 
             list(Duke_Out   = DEG.Duke.Out.UP$gene, 
                  S.Mira_Out = DEG.SMira.Out.UP$gene,
                  S.Shona_Out = DEG.SShona.Out.UP$gene), 
             scaled = FALSE, col = "black", fill = c("#E69F00","#56B4E9", "#009E73"),main = "Out of Infection - Up",
             euler.d = FALSE,lty = 1 , cex = 0.9, lwd = 1, cat.cex = 1, sigdigs = 1,
             print.mode=c("raw","percent"), height = 20, width = 20, cat.pos = c(-20,20,-180), cat.dist = c(0.06, 0.06, 0.04))
             
Venn.OutInf.DEG.DOWN <- venn.diagram(filename = NULL, 
             list(Duke_Out = DEG.Duke.Out.DOWN$gene, 
                  S.Mira_Out = DEG.SMira.Out.DOWN$gene,
                  S.Shona_Out = DEG.SShona.Out.DOWN$gene), 
             scaled = FALSE, col = "black", fill = c("#E69F00","#56B4E9", "#009E73"), main = "Out of Infection - Down", 
             euler.d = FALSE,lty = 1 , cex = 0.9, lwd = 1, cat.cex = 1, sigdigs = 1,
             print.mode=c("raw","percent"), height = 20, width = 20, cat.pos = c(-20,20,-180), cat.dist = c(0.06, 0.06, 0.04))
             
             
Venn.Inf.DEG.UP <- venn.diagram(filename = NULL, 
             list(Duke_Inf = DEG.Duke.Inf.UP$gene, 
                  S.Mira_Inf = DEG.SMira.Inf.UP$gene,
                  S.Shona_Inf = DEG.SShona.Inf.UP$gene), 
             scaled = FALSE, col = "black", fill = c("#E69F00","#56B4E9", "#009E73"), main = "Infection - Up",  
             euler.d = FALSE,lty = 1 , cex = 0.9, lwd = 1, cat.cex = 1, sigdigs = 1,
             print.mode=c("raw","percent"), height = 20, width = 20, cat.pos = c(-20,20,-180), cat.dist = c(0.06, 0.06, 0.04))
             

Venn.Inf.DEG.DOWN <- venn.diagram(filename = NULL, 
             list(Duke_Inf = DEG.Duke.Inf.DOWN$gene, 
                  S.Mira_Inf = DEG.SMira.Inf.DOWN$gene,
                  S.Shona_Inf = DEG.SShona.Inf.DOWN$gene), 
             scaled = FALSE, col = "black", fill = c("#E69F00","#56B4E9", "#009E73"), main = "Infection - Down",  
             euler.d = FALSE,lty = 1 , cex = 0.9, lwd = 1, cat.cex = 1, sigdigs = 1,
             print.mode=c("raw","percent"), height = 20, width = 20, cat.pos = c(-20,20,-180), cat.dist = c(0.06, 0.06, 0.04))
             


#DEGs but only seleceted by padj. <0.05:
Venn.OutInf.DEG_p.UP <- venn.diagram(filename = NULL,
             list(Duke_Out= DEG_p.Duke.Out.UP$gene,
                  SMira_Out = DEG_p.SMira.Out.UP$gene,
                  SShona_Out = DEG_p.SShona.Out.UP$gene),
             scaled = FALSE, col = "black", fill = c("#E69F00","#56B4E9", "#009E73"),main = "Out of Infection - Up - any log2FC",
             euler.d = FALSE,lty = 1 , cex = 0.9, lwd = 1, cat.cex = 1, sigdigs = 1,
             print.mode=c("raw","percent"), height = 20, width = 20, cat.pos = c(-20,20,-180), cat.dist = c(0.06, 0.06, 0.04))

Venn.OutInf.DEG_p.DOWN <- venn.diagram(filename = NULL,
             list(Duke_Out = DEG_p.Duke.Out.DOWN$gene,
                  SMira_Out = DEG_p.SMira.Out.DOWN$gene,
                  SShona_Out = DEG_p.SShona.Out.DOWN$gene),
             scaled = FALSE, col = "black", fill = c("#E69F00","#56B4E9", "#009E73"), main = "Out of Infection - Down - any log2FC",
             euler.d = FALSE,lty = 1 , cex = 0.9, lwd = 1, cat.cex = 1, sigdigs = 1,
             print.mode=c("raw","percent"), height = 20, width = 20, cat.pos = c(-20,20,-180), cat.dist = c(0.06, 0.06, 0.04))


Venn.Inf.DEG_p.UP <- venn.diagram(filename = NULL,
             list(Duke_Inf = DEG_p.Duke.Inf.UP$gene,
                  SMira_Inf = DEG_p.SMira.Inf.UP$gene,
                  SShona_Inf = DEG_p.SShona.Inf.UP$gene),
             scaled = FALSE, col = "black", fill = c("#E69F00","#56B4E9", "#009E73"), main = "Infection - Up - any log2FC",
             euler.d = FALSE,lty = 1 , cex = 0.9, lwd = 1, cat.cex = 1, sigdigs = 1,
             print.mode=c("raw","percent"), height = 20, width = 20, cat.pos = c(-20,20,-180), cat.dist = c(0.06, 0.06, 0.04))


Venn.Inf.DEG_p.DOWN <- venn.diagram(filename = NULL,
             list(Duke_Inf = DEG_p.Duke.Inf.DOWN$gene,
                  SMira_Inf = DEG_p.SMira.Inf.DOWN$gene,
                  SShona_Inf = DEG_p.SShona.Inf.DOWN$gene),
             scaled = FALSE, col = "black", fill = c("#E69F00","#56B4E9", "#009E73"), main = "Infection - Down - any log2FC",
             euler.d = FALSE,lty = 1 , cex = 0.9, lwd = 1, cat.cex = 1, sigdigs = 1,
             print.mode=c("raw","percent"), height = 20, width = 20, cat.pos = c(-20,20,-180), cat.dist = c(0.06, 0.06, 0.04))
             


grid.arrange(gTree(children = Venn.OutInf.DEG.UP),  gTree(children = Venn.OutInf.DEG.DOWN),
             gTree(children = Venn.OutInf.DEG_p.UP),  gTree(children = Venn.OutInf.DEG_p.DOWN), 
             gTree(children = Venn.Inf.DEG.UP), gTree(children = Venn.Inf.DEG.DOWN), 
             gTree(children = Venn.Inf.DEG_p.UP), gTree(children = Venn.Inf.DEG_p.DOWN),
             top = "Nro. of Genes with a padj.<0.05 & |log2FC| > 1      -      Nro. of Genes with a padj.<0.05",
             ncol = 4, nrow = 2, widths =unit(c(80,80,80,80), c("mm","mm","mm","mm")), heights = unit(c(80,80), c("mm","mm")))
```

## Merging DEGs with their gene_description, A.thalina_GeneID, and potatov404geneID:

```{r}
go_annotations = read.table(file = './analysis-data/DM_1-3_516_R44_potato.v6.1.working_models.go.txt', sep = '\t', header = FALSE)
go_annotations = go_annotations %>%
  mutate(gene = str_remove(V2, '\\.\\d+$')) %>%
  select(V2, gene, V5, V6) %>%
  rename(transcript_id = V2, GOterm = V5, TAIR = V6)
```

```{r}
func_annotations = read.table(file = './analysis-data/DM_1-3_516_R44_potato.v6.1.working_models.func_anno.txt', sep = '\t', header = FALSE)
func_annotations = func_annotations %>%
  mutate(gene = str_remove(V1, '\\.\\d+$')) %>%
  rename(transcript_id = V1, name = V2)
```

Data where the descriptions, Athaliana_geneID and v404 link of each gene
is saved:

NOTE: THIS IS COMMENTED OUT BECAUSE IT'S THE V4 STUFF (WE NEED TO UNCOMMENT IT OUT AFTER FOR COMPARISON PURPOSES)

```{r}
# #Genes ID / Names:
# load(file = "./analysis-data/potv6_Ath_potv404.RData")
# #Table: potv6_Ath_potv404
# colnames(potv6_Ath_potv404)[3] <- "gene"
# 
# #GeneID.GeneDescrip <- unique(potv6_Ath_potv404
# #head(GeneID.GeneDescrip)
# head(potv6_Ath_potv404)
```

Remember that GeneID_potv6 was linked to Athaliana_ID, and it was linked
to pot404: GeneID_potv6 -\> Athaliana_ID \<- pot404

```{r}
go_and_func_enrichment <- function (.data) {
  merge(merge(.data, go_annotations, by="gene", all.x = T), func_annotations, by="transcript_id", all.x = T) %>%
    select(-gene.y) %>%
    rename(gene = gene.x)
}
```

Adding annotation names to the result of the comparison:

```{r}
#Adding the annotation to all the DEG_p:
# DEG_p.Duke.Inf.annotation<- merge(DEG_p.Duke.Inf, potv6_Ath_potv404, by ="gene", all.x = T)
# DEG_p.Duke.Out.annotation<- merge(DEG_p.Duke.Out, potv6_Ath_potv404, by ="gene", all.x = T)
# 
# DEG_p.SMira.Inf.annotation<- merge(DEG_p.SMira.Inf, potv6_Ath_potv404, by ="gene", all.x = T)
# DEG_p.SMira.Out.annotation<- merge(DEG_p.SMira.Out, potv6_Ath_potv404, by ="gene", all.x = T)
# 
# DEG_p.SShona.Inf.annotation<- merge(DEG_p.SShona.Inf, potv6_Ath_potv404, by ="gene", all.x = T)
# DEG_p.SShona.Out.annotation<- merge(DEG_p.SShona.Out, potv6_Ath_potv404, by ="gene", all.x = T)
DEG_p.Duke.Inf.annotation<- go_and_func_enrichment(DEG_p.Duke.Inf)
DEG_p.Duke.Out.annotation<- go_and_func_enrichment(DEG_p.Duke.Out)

DEG_p.SMira.Inf.annotation<- go_and_func_enrichment(DEG_p.SMira.Inf)
DEG_p.SMira.Out.annotation<- go_and_func_enrichment(DEG_p.SMira.Out)

DEG_p.SShona.Inf.annotation<- go_and_func_enrichment(DEG_p.SShona.Inf)
DEG_p.SShona.Out.annotation<- go_and_func_enrichment(DEG_p.SShona.Out)


# Adding the annotation to all the DEGs:
DEG.Duke.Inf.annotation <- go_and_func_enrichment(DEG.Duke.Inf)
DEG.Duke.Out.annotation <- go_and_func_enrichment(DEG.Duke.Out)

DEG.SMira.Inf.annotation <- go_and_func_enrichment(DEG.SMira.Inf)
DEG.SMira.Out.annotation <- go_and_func_enrichment(DEG.SMira.Out)

DEG.SShona.Inf.annotation <- go_and_func_enrichment(DEG.SShona.Inf)
DEG.SShona.Out.annotation <- go_and_func_enrichment(DEG.SShona.Out)

# Adding the annotation to the comparison of TOL vs Duke:
DEG.SMira_Duke.Out.annotation  <- go_and_func_enrichment(DEG.SMira_vs_Duke.Out)
DEG.SMira_Duke.Inf.annotation  <- go_and_func_enrichment(DEG.SMira_vs_Duke.Inf)

DEG.SShona_Duke.Out.annotation <- go_and_func_enrichment(DEG.SShona_vs_Duke.Out)
DEG.SShona_Duke.Inf.annotation <- go_and_func_enrichment(DEG.SShona_vs_Duke.Inf)

# Adding the annotation to the "res":
res.Duke.Inf.tb   <- res.Duke.Inf %>% to_tibble()
res.Duke.Out.tb   <-  res.Duke.Out %>% to_tibble()
res.SMira.Inf.tb  <-  res.SMira.Inf %>% to_tibble()
res.SMira.Out.tb  <-  res.SMira.Out %>% to_tibble()
res.SShona.Inf.tb <-  res.SShona.Inf %>% to_tibble()
res.SShona.Out.tb <-  res.SShona.Out %>% to_tibble()

res.Duke.Inf.annotation   <- go_and_func_enrichment(res.Duke.Inf.tb)
res.Duke.Out.annotation   <- go_and_func_enrichment(res.Duke.Out.tb)
res.SMira.Inf.annotation  <- go_and_func_enrichment(res.SMira.Inf.tb)
res.SMira.Out.annotation  <- go_and_func_enrichment(res.SMira.Out.tb)
res.SShona.Inf.annotation <- go_and_func_enrichment(res.SShona.Inf.tb)
res.SShona.Out.annotation <- go_and_func_enrichment(res.SShona.Out.tb)

# Adding the annotation to the "res" in the comparison of each TOL vs Duke, in Infected and Outside infected area:
res.SMira_vs_Duke.Out.tb <- res.SMira_vs_Duke.Out %>% to_tibble()
res.SMira_vs_Duke.Inf.tb <- res.SMira_vs_Duke.Inf %>% to_tibble()
res.SShona_vs_Duke.Out.tb<- res.SShona_vs_Duke.Out %>% to_tibble()
res.SShona_vs_Duke.Inf.tb<- res.SShona_vs_Duke.Inf %>% to_tibble()

res.SMira_vs_Duke.Out.annotation  <- go_and_func_enrichment(res.SMira_vs_Duke.Out.tb)
res.SMira_vs_Duke.Inf.annotation  <- go_and_func_enrichment(res.SMira_vs_Duke.Inf.tb)
res.SShona_vs_Duke.Out.annotation <- go_and_func_enrichment(res.SShona_vs_Duke.Out.tb)
res.SShona_vs_Duke.Inf.annotation <- go_and_func_enrichment(res.SShona_vs_Duke.Inf.tb)
```

## Selecting DEG and DEG_p that are only in the tolerant but not in the susceptible after 48 of infection:

(1) Selecting DEGs in the intersection of TOLs but not in Duke, with annotation(\|log2FC\| \> 1 & padj. \< 0.05)

```{r}
# Gene list in "out of infection region" in both varieties:
OnlyTol.Out.DEG.up.genes <- setdiff(intersect(DEG.SMira.Out.UP$gene, DEG.SShona.Out.UP$gene), DEG.Duke.Out.UP$gene)
OnlyTol.Out.DEG.up.genes <- as.data.frame(OnlyTol.Out.DEG.up.genes)
colnames(OnlyTol.Out.DEG.up.genes)[1] <- "gene"


OnlyTol.Out.DEG.down.genes <- setdiff(intersect(DEG.SMira.Out.DOWN$gene, DEG.SShona.Out.DOWN$gene), DEG.Duke.Out.DOWN$gene)
OnlyTol.Out.DEG.down.genes <- as.data.frame(OnlyTol.Out.DEG.down.genes)
colnames(OnlyTol.Out.DEG.down.genes)[1] <- "gene"


# Adding gene expression:
indicesToRemove <- c(1, 3, 5, 6, 7)

OnlyTol.Out.DEG.up.annotation <- merge(merge(OnlyTol.Out.DEG.up.genes,
                                 unique(DEG.SMira.Out.annotation[,-indicesToRemove]), by = "gene", all.x=T),
                                 unique(DEG.SShona.Out.annotation[,-indicesToRemove]), by = "gene", all.x=T, suffixes = c('_SMira', '_SShona'))


OnlyTol.Out.DEG.down.annotation <- merge(merge(OnlyTol.Out.DEG.down.genes,
                                   unique(DEG.SMira.Out.annotation[,-indicesToRemove]), by = "gene", all.x =T),
                                   unique(DEG.SShona.Out.annotation[,-indicesToRemove]), by = "gene", all.x =T, suffixes = c('_SMira', '_SShona'))


#Gene list in "area of infection region" in both varieties:
OnlyTol.Inf.DEG.up.genes <- setdiff(intersect(DEG.SMira.Inf.UP$gene, DEG.SShona.Inf.UP$gene),DEG.Duke.Inf.UP$gene)
OnlyTol.Inf.DEG.up.genes <- as.data.frame(OnlyTol.Inf.DEG.up.genes)
colnames(OnlyTol.Inf.DEG.up.genes)[1] <- "gene"

OnlyTol.Inf.DEG.down.genes <- setdiff(intersect(DEG.SMira.Inf.DOWN$gene, DEG.SShona.Inf.DOWN$gene),DEG.Duke.Inf.DOWN$gene)
OnlyTol.Inf.DEG.down.genes <- as.data.frame(OnlyTol.Inf.DEG.down.genes)
colnames(OnlyTol.Inf.DEG.down.genes)[1] <- "gene"


#Adding gene expression:
OnlyTol.Inf.DEG.up.annotation <- merge(merge(OnlyTol.Inf.DEG.up.genes,
                          unique(DEG.SMira.Inf.annotation[,-indicesToRemove]), by = "gene", all.x=T),
                          unique(DEG.SShona.Inf.annotation[,-indicesToRemove]), by = "gene", all.x=T, suffixes = c('_SMira', '_SShona'))

OnlyTol.Inf.DEG.down.annotation <- merge(merge(OnlyTol.Inf.DEG.down.genes,
                          unique(DEG.SMira.Inf.annotation[,-indicesToRemove]), by = "gene", all.x=T),
                          unique(DEG.SShona.Inf.annotation[,-indicesToRemove]), by = "gene", all.x=T, suffixes = c('_SMira', '_SShona'))
```

(2)Selecting DEG (\|log2FC\| \> 1 & padj. \< 0.05) not adding
annotation:

```{r}
#Intersect Sarpo Mira & Sarpo Shona but not Duke
View(as.data.frame(setdiff(intersect(DEG.SMira.Inf.UP$gene,DEG.SShona.Inf.UP$gene),DEG.Duke.Inf.UP$gene)))
View(as.data.frame(setdiff(intersect(DEG.SMira.Inf.DOWN$gene,DEG.SShona.Inf.DOWN$gene),DEG.Duke.Inf.DOWN$gene)))

#DEG only Sarpo Mira
View(as.data.frame(setdiff(setdiff(DEG.SMira.Inf.UP$gene,DEG.SShona.Inf.UP$gene),DEG.Duke.Inf.UP$gene)))
View(as.data.frame(setdiff(setdiff(DEG.SMira.Inf.DOWN$gene,DEG.SShona.Inf.DOWN$gene),DEG.Duke.Inf.DOWN$gene)))

#DEG only Sarpo Shona
View(as.data.frame(setdiff(setdiff(DEG.SShona.Inf.UP$gene,DEG.SMira.Inf.UP$gene),DEG.Duke.Inf.UP$gene)))
View(as.data.frame(setdiff(setdiff(DEG.SShona.Inf.DOWN$gene,DEG.SMira.Inf.DOWN$gene),DEG.Duke.Inf.DOWN$gene)))
```

(3)Selecting DEG_p (& padj. \< 0.05)

```{r}
OnlyTol.Out.DEG_p.up.genes <- setdiff(intersect(DEG_p.SMira.Out.UP$gene, DEG_p.SShona.Out.UP$gene),DEG_p.Duke.Out.UP$gene)
OnlyTol.Out.DEG_p.up.genes <- as.data.frame(OnlyTol.Out.DEG_p.up.genes)
colnames(OnlyTol.Out.DEG_p.up.genes)[1] <- "gene"


OnlyTol.Out.DEG_p.down.genes <- setdiff(intersect(DEG_p.SMira.Out.DOWN$gene, DEG_p.SShona.Out.DOWN$gene),DEG_p.Duke.Out.DOWN$gene)
OnlyTol.Out.DEG_p.down.genes <- as.data.frame(OnlyTol.Out.DEG_p.down.genes)
colnames(OnlyTol.Out.DEG_p.down.genes)[1] <- "gene"


#Adding gene expression:
OnlyTol.Out.DEG_p.up.annotation <- merge(merge(OnlyTol.Out.DEG_p.up.genes,
                                             unique(DEG_p.SMira.Out.annotation[,-indicesToRemove]), by = "gene", all.x =T),
                                       unique(DEG_p.SShona.Out.annotation[,-indicesToRemove]), by = "gene", all.x =T) #269 DEG_ps


OnlyTol.Out.DEG_p.down.annotation <- merge(merge(OnlyTol.Out.DEG_p.down.genes,
                                               unique(DEG_p.SMira.Out.annotation[,-indicesToRemove]), by = "gene", all.x =T),
                                         unique(DEG_p.SShona.Out.annotation[,-indicesToRemove]), by = "gene", all.x =T) #44 DEG_ps



#Gene list in "area of infection region" in both varieties:
OnlyTol.Inf.DEG_p.up.genes <- setdiff(intersect(DEG_p.SMira.Inf.UP$gene, DEG_p.SShona.Inf.UP$gene),DEG_p.Duke.Inf.UP$gene)
OnlyTol.Inf.DEG_p.up.genes <- as.data.frame(OnlyTol.Inf.DEG_p.up.genes)
colnames(OnlyTol.Inf.DEG_p.up.genes)[1] <- "gene"

OnlyTol.Inf.DEG_p.down.genes <- setdiff(intersect(DEG_p.SMira.Inf.DOWN$gene, DEG_p.SShona.Inf.DOWN$gene),DEG_p.Duke.Inf.DOWN$gene)
OnlyTol.Inf.DEG_p.down.genes <- as.data.frame(OnlyTol.Inf.DEG_p.down.genes)
colnames(OnlyTol.Inf.DEG_p.down.genes)[1] <- "gene"


#Adding gene expression:
OnlyTol.Inf.DEG_p.up.annotation <- merge(merge(OnlyTol.Inf.DEG_p.up.genes,
                                       unique(DEG_p.SMira.Inf.annotation[,-indicesToRemove]), by = "gene", all.x =T),
                                 unique(DEG_p.SShona.Inf.annotation[,-indicesToRemove]), by = "gene", all.x =T) #291 DEG_p_p

OnlyTol.Inf.DEG_p.down.annotation <- merge(merge(OnlyTol.Inf.DEG_p.down.genes,
                                         unique(DEG_p.SMira.Inf.annotation[,-indicesToRemove]), by = "gene", all.x =T),
                                   unique(DEG_p.SShona.Inf.annotation[,-indicesToRemove]), by = "gene", all.x =T) #114 DEG_p_p

```

## Selecting DEG in the intersection Inf48 & Out48, in each variety:

```{r}
#Gene list in "out of infection region" in both varieties:
SMira.Inf_Out.DEG.up.genes <- as.data.frame(intersect(DEG.SMira.Inf.UP$gene, DEG.SMira.Out.UP$gene))
colnames(SMira.Inf_Out.DEG.up.genes)[1] <- "gene"

SShona.Inf_Out.DEG.up.genes <- as.data.frame(intersect(DEG.SShona.Inf.UP$gene, DEG.SShona.Out.UP$gene))
colnames(SShona.Inf_Out.DEG.up.genes)[1] <- "gene"

Duke.Inf_Out.DEG.up.genes <- as.data.frame(intersect(DEG.Duke.Inf.UP$gene, DEG.Duke.Out.UP$gene))
colnames(Duke.Inf_Out.DEG.up.genes)[1] <- "gene"

SMira.Inf_Out.DEG.DOWN.genes <- as.data.frame(intersect(DEG.SMira.Inf.DOWN$gene, DEG.SMira.Out.DOWN$gene))
colnames(SMira.Inf_Out.DEG.DOWN.genes)[1] <- "gene"

SShona.Inf_Out.DEG.DOWN.genes <- as.data.frame(intersect(DEG.SShona.Inf.DOWN$gene, DEG.SShona.Out.DOWN$gene))
colnames(SShona.Inf_Out.DEG.DOWN.genes)[1] <- "gene"

Duke.Inf_Out.DEG.DOWN.genes <- as.data.frame(intersect(DEG.Duke.Inf.DOWN$gene, DEG.Duke.Out.DOWN$gene))
colnames(Duke.Inf_Out.DEG.DOWN.genes)[1] <- "gene"


#Adding gene expression:

SMira.Inf_Out.DEG.up.exp <- merge(merge(SMira.Inf_Out.DEG.up.genes,
                            unique(DEG.SMira.Inf.annotation[,-indicesToRemove]), by = "gene", all.x =T),
                            unique(DEG.SMira.Out.annotation[,-indicesToRemove]), by = "gene", all.x =T, suffixes = c('_SMira_Inf', '_SMira_Out'))

SShona.Inf_Out.DEG.up.exp <- merge(merge(SShona.Inf_Out.DEG.up.genes,
                             unique(DEG.SShona.Inf.annotation[,-indicesToRemove]), by = "gene", all.x =T),
                             unique(DEG.SShona.Out.annotation[,-indicesToRemove]), by = "gene", all.x =T, suffixes = c('_SShona_Inf', '_SShona_Out'))

Duke.Inf_Out.DEG.up.exp <- merge(merge(Duke.Inf_Out.DEG.up.genes,
                           unique(DEG.Duke.Inf.annotation[,-indicesToRemove]), by = "gene", all.x =T),
                           unique(DEG.Duke.Out.annotation[,-indicesToRemove]), by = "gene", all.x =T, suffixes = c('_Duke_Inf', '_Duke_Out'))


SMira.Inf_Out.DEG.DOWN.exp <- merge(merge(SMira.Inf_Out.DEG.DOWN.genes,
                                  unique(DEG.SMira.Inf.annotation[,-indicesToRemove]), by = "gene", all.x =T),
                                  unique(DEG.SMira.Out.annotation[,-indicesToRemove]), by = "gene", all.x =T, suffixes = c('_SMira_Inf', '_SMira_Out'))

SShona.Inf_Out.DEG.DOWN.exp <- merge(merge(SShona.Inf_Out.DEG.DOWN.genes,
                                         unique(DEG.SShona.Inf.annotation[,-indicesToRemove]), by = "gene", all.x =T),
                                   unique(DEG.SShona.Out.annotation[,-indicesToRemove]), by = "gene", all.x =T, suffixes = c('_SShona_Inf', '_SShona_Out'))

Duke.Inf_Out.DEG.DOWN.exp <- merge(merge(Duke.Inf_Out.DEG.DOWN.genes,
                                       unique(DEG.Duke.Inf.annotation[,-indicesToRemove]), by = "gene", all.x =T),
                                 unique(DEG.Duke.Out.annotation[,-indicesToRemove]), by = "gene", all.x =T, suffixes = c('_Duke_Inf', '_Duke_Out'))
```

## DEG from the comparison of each TOL vs Duke, in infected or outinfected area:

NOTE: temporarily disabled - need to rework the indices for v6 use!

```{r}
# TOL_vs_Duke.genes <- unique(union(union(DEG.SMira_vs_Duke.Inf$gene, DEG.SMira_vs_Duke.Out$gene),
#                                   union(DEG.SShona_vs_Duke.Inf$gene,DEG.SShona_vs_Duke.Out$gene))) #8683 DEGs in total
# 
# TOL_vs_Duke.annotation <- unique(rbind(rbind(rbind(
#                              subset(unique(res.SMira_vs_Duke.Inf.annotation[,c(1,10,8)]), gene %in% TOL_vs_Duke.genes),
#                              subset(unique(res.SMira_vs_Duke.Out.annotation[,c(1,10,8)]), gene %in% TOL_vs_Duke.genes)),
#                              subset(unique(res.SShona_vs_Duke.Inf.annotation[,c(1,10,8)]), gene %in% TOL_vs_Duke.genes)),
#                              subset(unique(res.SShona_vs_Duke.Out.annotation[,c(1,10,8)]), gene %in% TOL_vs_Duke.genes))) #8683 DEGs in total
# 
# TOL_vs_Duke.Inf.Out <- merge(merge(merge(merge(TOL_vs_Duke.annotation,
#                              subset(as.data.frame(res.SMira_vs_Duke.Inf.tb[,c(1,3,7)]), gene %in% TOL_vs_Duke.genes),by = "gene", all.x =T),
#                              subset(as.data.frame(res.SMira_vs_Duke.Out.tb[,c(1,3,7)]), gene %in% TOL_vs_Duke.genes),by = "gene", all.x =T),
#                              subset(as.data.frame(res.SShona_vs_Duke.Inf.tb[,c(1,3,7)]), gene %in% TOL_vs_Duke.genes),by = "gene", all.x =T),
#                              subset(as.data.frame(res.SShona_vs_Duke.Out.tb[,c(1,3,7)]), gene %in% TOL_vs_Duke.genes),by = "gene", all.x =T)
# 
# colnames(TOL_vs_Duke.Inf.Out) <- c("gene","Gene_Name","Athaliana_geneID",
#                                    "SMira.Inf.LFC","SMira.Inf.padj","SMira.Out.LFC","SMira.Out.padj", 
#                                    "SShona.Inf.LFC","SShona.Inf.padj","SShona.Out.LFC","SShona.Out.padj")

```

## Plotting the normalized gene counts across the different time points.

It is a graph for an individual gene.

```{r, echo=FALSE, fig.height=5, fig.width=20}
plotCounts(dds_Blight, gene= "Soltu.DM.01G027350", intgroup=c("variety","treatment"))
```

## Selection of the top 20:

NOTE: temporarily disabled - need to rework the indices for v6 use!

Sarpo Mira (Inf/Out vs Mock):

```{r, echo=FALSE, fig.height=5, fig.width=20}
# DEG.SMira.Inf.annotation.up <- unique(subset(DEG.SMira.Inf.annotation,log2FoldChange > 1)[,c(1,3,7,8,10)])
# DEG.SMira.Out.annotation.up <- unique(subset(DEG.SMira.Out.annotation,log2FoldChange > 1)[,c(1,3,7,8,10)])
# 
# DEG.SMira.Inf.annotation.down <- unique(subset(DEG.SMira.Inf.annotation,log2FoldChange < -1)[,c(1,3,7,8,10)])
# DEG.SMira.Out.annotation.down <- unique(subset(DEG.SMira.Out.annotation,log2FoldChange < -1)[,c(1,3,7,8,10)])
```

Duke (Inf/Out vs Mock):

```{r, echo=FALSE, fig.height=5, fig.width=20}
# DEG.Duke.Inf.annotation.up <- unique(subset(DEG.Duke.Inf.annotation,log2FoldChange > 1)[,c(1,3,7,8,10)])
# DEG.Duke.Out.annotation.up <- unique(subset(DEG.Duke.Out.annotation,log2FoldChange > 1)[,c(1,3,7,8,10)])
# 
# DEG.Duke.Inf.annotation.down <- unique(subset(DEG.Duke.Inf.annotation,log2FoldChange < -1)[,c(1,3,7,8,10)])
# DEG.Duke.Out.annotation.down <- unique(subset(DEG.Duke.Out.annotation,log2FoldChange < -1)[,c(1,3,7,8,10)])
```

Sarpo Mira vs Duke, Inf48

```{r}
# DEG.SMira_vs_Duke.Inf.annotation.up <- unique(subset(DEG.SMira_Duke.Inf.annotation, log2FoldChange > 1)[,c(1,3,7,8,10)])
# DEG.SMira_vs_Duke.Inf.annotation.down <- unique(subset(DEG.SMira_Duke.Inf.annotation, log2FoldChange < -1)[,c(1,3,7,8,10)])
```

Sarpo Mira vs Duke, OutInf48h

```{r}
# DEG.SMira_vs_Duke.Out.annotation.up <- unique(subset(DEG.SMira_Duke.Out.annotation, log2FoldChange > 1)[,c(1,3,7,8,10)])
# DEG.SMira_vs_Duke.Out.annotation.down <- unique(subset(DEG.SMira_Duke.Out.annotation, log2FoldChange < -1)[,c(1,3,7,8,10)])
```

Sarpo Shona vs Duke, Inf48h

```{r}
# DEG.SShona_vs_Duke.Inf.annotation.up <- unique(subset(DEG.SShona_Duke.Inf.annotation, log2FoldChange > 1)[,c(1,3,7,8,10)])
# DEG.SShona_vs_Duke.Inf.annotation.down <- unique(subset(DEG.SShona_Duke.Inf.annotation, log2FoldChange < -1)[,c(1,3,7,8,10)])
```

Sarpo Mira vs Duke, OutInf48h

```{r}
# DEG.SShona_vs_Duke.Out.annotation.up <- unique(subset(DEG.SShona_Duke.Out.annotation, log2FoldChange > 1)[,c(1,3,7,8,10)])
# DEG.SShona_vs_Duke.Out.annotation.down <- unique(subset(DEG.SShona_Duke.Out.annotation, log2FoldChange < -1)[,c(1,3,7,8,10)])
```

## Saving data

```{r}
#For gProfiler2, without annotation:

#save(DEG.Duke.Inf.UP, DEG.Duke.Inf.DOWN, DEG.Duke.Out.UP, DEG.Duke.Out.DOWN,
#     DEG.SMira.Inf.UP, DEG.SMira.Inf.DOWN, DEG.SMira.Out.UP, DEG.SMira.Out.DOWN,
#     DEG.SShona.Inf.UP, DEG.SShona.Inf.DOWN, DEG.SShona.Out.UP, DEG.SShona.Out.DOWN, file = "2.DESeq2_output_48h/DEG_48h_vs_Mock_UpDown.RData")


#save(DEG_p.Duke.Inf.UP, DEG_p.Duke.Inf.DOWN, DEG_p.Duke.Out.UP, DEG_p.Duke.Out.DOWN,
#     DEG_p.SMira.Inf.UP, DEG_p.SMira.Inf.DOWN, DEG_p.SMira.Out.UP, DEG_p.SMira.Out.DOWN,
#     DEG_p.SShona.Inf.UP, DEG_p.SShona.Inf.DOWN, DEG_p.SShona.Out.UP, DEG_p.SShona.Out.DOWN, file = "2.DESeq2_output_48h/DEG_p_48h_vs_Mock_UpDown.RData")





#DEG with their annotation(gene_name, A.thaliana_ID, GOterm) to later select DEGs in base of their linked GO terms:
#save(DEG.Duke.Inf.annotation,DEG.Duke.Out.annotation,
#     DEG.SMira.Inf.annotation,DEG.SMira.Out.annotation,
#     DEG.SShona.Inf.annotation,DEG.SShona.Out.annotation, file ="2.DESeq2_output_48h/DEG_48h_vs_Mock.Annotation_.RData")

#DEG with without GOterms to observe only the gene_name and the A.thaliana_ID. 
#It generate a smaller tables than the previous one to observe better which genes are DEGs:
#write.csv(unique(DEG.Duke.Inf.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_1Duke_Inf48h_vs_Mock_GeneFuntion_notGO.csv")
#write.csv(unique(DEG.Duke.Out.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_1Duke_Out48h_vs_Mock_GeneFuntion_notGO.csv")

#write.csv(unique(DEG.SMira.Inf.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_1SMira_Inf48h_vs_Mock_GeneFuntion_notGO.csv")
#write.csv(unique(DEG.SMira.Out.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_1SMira_Out48h_vs_Mock_GeneFuntion_notGO.csv")

#write.csv(unique(DEG.SShona.Inf.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_1SShona_Inf48h_vs_Mock_GeneFuntion_notGO.csv")
#write.csv(unique(DEG.SShona.Out.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_1SShona_Out48h_vs_Mock_GeneFuntion_notGO.csv")


#To complete the expression of the genes in the tables that will be generated:
#save(res.Duke.Inf, res.Duke.Out, 
#     res.SMira.Inf, res.SMira.Out, 
#     res.SShona.Inf, res.SShona.Out, file = "2.DESeq2_output_48h/res.48h_vs_Mock.RData")

#save(res.Duke.Inf.annotation, res.Duke.Out.annotation, 
#     res.SMira.Inf.annotation, res.SMira.Out.annotation, 
#     res.SShona.Inf.annotation, res.SShona.Out.annotation, file = "2.DESeq2_output_48h/res.48h_vs_Mock.annotation.RData")


#Saving the DEGs from the comparison between each TOL vs Duke in infected or outinfected area:
#write.csv(TOL_vs_Duke.Inf.Out, file = "EachDEG.TOL_vs_Duke_Inf_Out.csv")


#DEG_p:
#save(DEG_p.Duke.Inf.annotation,DEG_p.Duke.Out.annotation,
#     DEG_p.SMira.Inf.annotation,DEG_p.SMira.Out.annotation,
#     DEG_p.SShona.Inf.annotation,DEG_p.SShona.Out.annotation, file ="2.DESeq2_output_48h/DEG_p_Annotation_48h_vs_Mock.RData")

#write.csv(unique(DEG_p.Duke.Inf.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_p_Duke_Inf48h_vs_Mock_GeneFuntion_notGO.csv")
#write.csv(unique(DEG_p.Duke.Out.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_p_Duke_Out48h_vs_Mock_GeneFuntion_notGO.csv")

#write.csv(unique(DEG_p.SMira.Inf.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_p_SMira_Inf48h_vs_Mock_GeneFuntion_notGO.csv")
#write.csv(unique(DEG_p.SMira.Out.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_p_SMira_Out48h_vs_Mock_GeneFuntion_notGO.csv")

#write.csv(unique(DEG_p.SShona.Inf.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_p_SShona_Inf48h_vs_Mock_GeneFuntion_notGO.csv")
#write.csv(unique(DEG_p.SShona.Out.annotation[,c(1,3,7,8,10)]), file = "2.DESeq2_output_48h/DEG_p_SShona_Out48h_vs_Mock_GeneFuntion_notGO.csv")

#Saving normalized counts: (from "Norm.Counts <- counts(dds_Blight, normalized=TRUE)")
#write.csv(Norm.Counts, file= "NormalizedCounts.csv")

```
