---
title: "Transcriptomic analysis of potato samples infected with P. infestans - DE analysis"
author: "Liam Cremona, Niall Dunne, Ossama Edbali - Patricia Ponce"
date: "2022-2023"
output:
  html_document:
    self_contained: false
fontsize: 15pt
editor_options:
  markdown: 
    wrap: 72
---

> This script has been apadted from Patricia Ponce

# Introduction

The data comes from potato leaves infected with Phythophtora infestans
for 48 hours using a detached infection assay. Before this analysis the
reads were mapped to the potato genome v6 with STAR, then counted with
HTSeq. Three potato varieties were infected: Sarpo Shona (Tolerant),
Sarpo Mira (Tolerant) and Duke of York (Susceptible).

## Purpose

This part of the transcriptomic analysis involve the
normalization of the counts by DESeq2. After that we plot a PCA to
visualize the distribution of the samples, then find which genes are
significantly different between varieties 48 hours after infection.
All the comparisons will be done with against the mock (Infected-Area vs
Mock or Out-infection-Area vs Mock).

## Input

- HTseq counts in the `quantified` directory.
- Annotation info from the potato genome v6.1 in "Z:/1.PotatoDroughtOut_genomev6/4.GO_analysis/gProfile_FileHC.GeneTransID.Name.GO/potv6_Ath_potv404.RData".

## Output

- "1.DESeq2_analysis_blight_infection48h.html" (After kniting)
- "1.DESeq2_analysis_blight_infection48h_files" --\> Inside the folder
"figure-html" are all the graphs produced with this code. (After
kniting) -All the files in the "2.DESeq2_output_48h" folder

# Setup

Path of the folder for the inputs and outputs:

```{r}
setwd("/rds/projects/l/leachlj-potato-qtl-project/M6_Area/potato-blight-rna-seq-pipeline")
```

Packages to employ for the analysis:

```{r message=FALSE, warning=FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("apeglm")
```

```{r}
BiocManager::install("DESeq2", force = TRUE)
```

```{r message=FALSE, warning=FALSE}
#this code works to install an older version of lasso2 from the archives
install.packages("https://cran.r-project.org/src/contrib/Archive/lasso2/lasso2_1.2-22.tar.gz", repos=NULL, type="source")
```

```{r message=FALSE, warning=FALSE}
BiocManager::install("DEGreport", force = TRUE)
```

```{r}
BiocManager::install("EnhancedVolcano", force = TRUE)
```

```{r}
library(apeglm)
library(DESeq2)
library(DEGreport)
library(EnhancedVolcano)
library(ggplot2)
library(pheatmap)
library(gplots)
library(grDevices)
library(RColorBrewer)
library(pheatmap)
library(dplyr)
library(tibble)
library(data.table)
library(gridExtra)
library(stringr)
library(VennDiagram)
```

```{r}
source('./helpers.R')
source('./plots.R')
source('./comparisons.R')
source('./degs.R')
```

```{r}
#specify this to enable plotting to work
options(bitmapType='cairo')
```

# Setup input analysis data directory

```{r}
# TODO
```

# Setup results directory

```{r}
if (!dir.exists("results")) {
  dir.create("results")
}

if (!dir.exists("results/different-treatments-within-varieties")) {
  dir.create("results/different-treatments-within-varieties")
}

if (!dir.exists("results/same-treatment-between-varieties")) {
  dir.create("results/same-treatment-between-varieties")
}
```

# Input configuration

```{r}
use_v4 = FALSE
```

# Prepare data

```{r}
directory = './quantified'
sampleFiles = grep('*count.txt', list.files(directory), value = TRUE)
sampleFilesR = sort(sampleFiles[startsWith(sampleFiles, 'R')])
sampleFilesA = sort(sampleFiles[startsWith(sampleFiles, 'A')])
sampleFiles = c(sampleFilesR, sampleFilesA)
sampleNames = c(
  'SMira_C1', 'SMira_C2', 'SMira_C3',
  'SMira_OI1', 'SMira_OI2', 'SMira_OI3',
  'Duke_C1', 'Duke_C2', 'Duke_C3',
  'Duke_OI1', 'Duke_OI2', 'Duke_OI3',
  'SShona_C1', 'SShona_C2', 'SShona_C3',
  'SShona_OI1', 'SShona_OI2', 'SShona_OI3',
  'SMira_I1', 'SMira_I2', 'SMira_I3',
  'Duke_I1', 'Duke_I2', 'Duke_I3',
  'SShona_I1', 'SShona_I2', 'SShona_I3',
  'SMira_M1', 'SMira_M2', 'SMira_M3',
  'Duke_M1', 'Duke_M2', 'Duke_M3',
  'SShona_M1', 'SShona_M2', 'SShona_M3'
)

sampleReplicate = rep(c(1, 2, 3), 12)

sampleVariety = c(
  rep('SMira', 6),
  rep('Duke', 6),
  rep('SShona', 6),
  rep('SMira', 3),
  rep('Duke', 3),
  rep('SShona', 3),
  rep('SMira', 3),
  rep('Duke', 3),
  rep('SShona', 3)
)

sampleTreatment = c(
  rep('Control', 3),
  rep('OutInf', 3),
  rep('Control', 3),
  rep('OutInf', 3),
  rep('Control', 3),
  rep('OutInf', 3),
  rep('Inf', 9),
  rep('Mock', 9)
)

sampleTable = data.frame(sampleNames= sampleNames, 
                          fileName = sampleFiles, 
                          replicate = sampleReplicate,
                          variety = sampleVariety,
                          treatment = sampleTreatment)

sampleTable$replicate = factor(sampleTable$replicate)
sampleTable$variety = factor(sampleTable$variety)
sampleTable$treatment = factor(sampleTable$treatment)

varieties <- list(
  Duke=list(
    type="susceptible"
  ),
  SMira=list(
    type="tolerant"
  ),
  SShona=list(
    type="tolerant"
  )
)

regulation_directions = c("UP", "DOWN")
```

# Run DESeq2 with all the samples

```{r}
dds_Blight <- DESeqDataSetFromHTSeqCount(
  sampleTable = sampleTable,
  directory = directory,
  design = ~variety + treatment + variety:treatment
)

# Removing genes whose counts is less than 10 in total around all the libraries
keep <- rowSums(counts(dds_Blight)) >= 10
dds_Blight <- dds_Blight[keep,]

dds_Blight$variety <- relevel(dds_Blight$variety, ref = c("Duke"))

# 1. Reference treatment: Mock
dds_Blight$treatment <- relevel(dds_Blight$treatment, ref = c("Mock"))
dds_Blight <- DESeq(dds_Blight)

# 2. Reference treatment: Control
dds_Blight$treatment <- relevel(dds_Blight$treatment, ref = c("Control"))
dds_Blight_varieties_mock <- DESeq(dds_Blight)
```

# Exploratory analysis

## Normalized counts

```{r}
Norm.Counts <- counts(dds_Blight, normalized=TRUE)
```

## Gene count distribution per samples in leaf

Distribution before normalization:

```{r}
boxplot(log2(counts(dds_Blight) + 1), range=0, las=2)
```

Distribution after normalization:

```{r}
boxplot(log2(counts(dds_Blight, normalized=TRUE) + 1), range=0, las=2)
```

## Principal Component Analysis (PCA) of the leaf samples

Variance stabilization transformation:

```{r}
vst_Blight <- vst(dds_Blight, blind=TRUE)
```

PCA of all leaf samples:

```{r, echo=FALSE, fig.height=7, fig.width=14}
# n = 29393
pcaVST <- plotPCA(vst_Blight, intgroup = c("treatment","variety"), returnData = TRUE, ntop = 29393)
percentVar <- round(100*attr(pcaVST, "percentVar"))

SusTol_all <- qplot(PC1, PC2, color=treatment, shape=variety, data=pcaVST) +
  geom_point(size=3)+
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))+
  labs(title="All leaf samples_n29393")+
  geom_point()+
  geom_text(aes(label=colnames(vst_Blight)),size = 3, vjust=1.5, hjust=1)


# n = 1000
pcaVST_n1000 <- plotPCA(vst_Blight, intgroup = c("treatment","variety"), returnData = TRUE, ntop = 1000)
percentVar_n1000 <- round(100*attr(pcaVST_n1000, "percentVar"))

SusTol_n1000 <- qplot(PC1, PC2, color=treatment, shape=variety, data=pcaVST_n1000) +
           geom_point(size=3)+
           xlab(paste0("PC1: ",percentVar_n1000[1],"% variance")) +
           ylab(paste0("PC2: ",percentVar_n1000[2],"% variance"))+
           labs(title="All leaf samples_n1000")+
           geom_point()+
           geom_text(aes(label=colnames(vst_Blight)),size = 3, vjust=1.5, hjust=1)

grid.arrange(SusTol_all, SusTol_n1000,
             ncol=2, widths=unit(c(160,160), c("mm","mm")), heights=unit(150, "mm"))
```

The distribution of the samples is very similar in both cases, by using
all the normalized genes or using only the 1000 most variable genes. All
replicates looks good, then it won't be necessary to remove any sample.

## Heatmap and clustering of the samples

Heatmap of all leaf samples, after removing bad biological replicates:

```{r}
sampleDists <- dist(t(assay(vst_Blight)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vst_Blight$variety, vst_Blight$treatment, sep="-")
colnames(sampleDistMatrix) <- NULL

colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)

pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)
```

# Setup comparisons

> Tutorial: <https://rstudio-pubs-static.s3.amazonaws.com/329027_593046fb6d7a427da6b2c538caf601e1.html>

Data to compare:

```{r}
resultsNames(dds_Blight) # input for "name" and "contrast"
resultsNames(dds_Blight_varieties_mock)
```

For the comparison: - Base_level: Duke / Mock - res: To get the pvalue
and lfcThreshold=0, - default lfcThreshold=0

For the DE analysis, genes with the absolute value of the log2folchange
more than 1 (\|log2FC\| \> 1) will be selected.

Pairwise comparisons between samples (Inf/OutInf vs Mock and tolerant vs Duke in Infect/OutInfect. area) from the tolerant variety:

```{r}
comparisons = list(
  res.Duke.Out = c("treatment_OutInf_vs_Mock"),
  res.Duke.Inf = c("treatment_Inf_vs_Mock"),
  res.Duke.Control = c("treatment_Control_vs_Mock"),
  
  res.SMira.Out = c("treatment_OutInf_vs_Mock","varietySMira.treatmentOutInf"),
  res.SMira.Inf = c("treatment_Inf_vs_Mock","varietySMira.treatmentInf"),
  res.SMira.Control = c("treatment_Control_vs_Mock","varietySMira.treatmentControl"),
  
  res.SShona.Out = c("treatment_OutInf_vs_Mock","varietySShona.treatmentOutInf"),
  res.SShona.Inf = c("treatment_Inf_vs_Mock","varietySShona.treatmentInf"),
  res.SShona.Control = c("treatment_Control_vs_Mock","varietySShona.treatmentControl"),
  
  res.SMira_vs_Duke.Out = c("variety_SMira_vs_Duke" ,"varietySMira.treatmentOutInf"),
  res.SMira_vs_Duke.Inf = c("variety_SMira_vs_Duke" ,"varietySMira.treatmentInf"),
  res.SMira_vs_Duke.Control = c("variety_SMira_vs_Duke" ,"varietySMira.treatmentControl"),
  
  res.SShona_vs_Duke.Out = c("variety_SShona_vs_Duke","varietySShona.treatmentOutInf"),
  res.SShona_vs_Duke.Inf = c("variety_SShona_vs_Duke","varietySShona.treatmentInf"),
  res.SShona_vs_Duke.Control = c("variety_SShona_vs_Duke" ,"varietySShona.treatmentControl")
)

comparisons_varieties_mock <- list(
  res.SMira_vs_Duke.Mock = c('variety_SMira_vs_Duke', 'varietySMira.treatmentMock'),
  res.SShona_vs_Duke.Mock = c('variety_SShona_vs_Duke', 'varietySShona.treatmentMock')
)

res <- create_comparisons(dds_Blight, comparisons)
res_varieties_mock <- create_comparisons(dds_Blight_varieties_mock, comparisons_varieties_mock)

res <- c(res, res_varieties_mock)
```

# Selecting Differentially Expressed Genes (DEGs)

```{r}
degs = create_degs(res)
```

# Get up and down regulated genes

```{r}
up_down_degs = create_up_down_degs(degs)
```

# Merging DEGs with their GO and functional annotations

```{r}
if (use_v4) {
  load(file = "./analysis-data/potv6_Ath_potv404.RData")
  colnames(potv6_Ath_potv404)[3] <- "gene"
} else {
  go_annotations = read.table(file = './analysis-data/DM_1-3_516_R44_potato.v6.1.working_models.go.txt', sep = '\t', header = FALSE)
  go_annotations = go_annotations %>%
    dplyr::mutate(gene = str_remove(V2, '\\.\\d+$')) %>%
    dplyr::select(V2, gene, V5, V6) %>%
    dplyr::rename(transcript_id = 1, GOterm = 3, TAIR = 4)
  
  func_annotations = read.table(file = './analysis-data/DM_1-3_516_R44_potato.v6.1.working_models.func_anno.txt', sep = '\t', header = FALSE)
  func_annotations = func_annotations %>%
    dplyr::mutate(gene = str_remove(V1, '\\.\\d+$')) %>%
    dplyr::rename(transcript_id = V1, name = V2)
}
```

```{r}
go_and_func_enrichment <- function (.data) {
  if (use_v4) {
    merge(.data, potv6_Ath_potv404, by ="gene", all.x = T)
  } else {
    merge(merge(.data, go_annotations, by="gene", all.x = T), func_annotations, by="transcript_id", all.x = T) %>%
      dplyr::select(-gene.y) %>%
      dplyr::rename(gene = gene.x)
  }
}
```

Adding annotation names to the result of the comparisons:

```{r}
annotated_degs <- lapply(degs, go_and_func_enrichment)
annotated_up_down_degs <- lapply(up_down_degs, go_and_func_enrichment)
```

# Comparing different treatments WITHIN varieties

## Volcano plots

```{r, echo=FALSE, fig.height=20, fig.width=14}
volcano_plots_from_comparisons(res,
                               filename = "results/different-treatments-within-varieties/volcano-plots.pdf",
                               ncol = 2,
                               nrow = 3,
                               widths = unit(c(160,160), c("mm","mm")),
                               heights = unit(c(150,150,150), c("mm","mm","mm")),
                               condition = function(name) { !grepl("_vs_", name) })
```

## Venn diagrams

```{r, echo=FALSE, fig.height=13, fig.width=13}
create_venn_diagrams_grid(
  up_down_degs,
  ncol = 2,
  nrow = 3,
  widths = unit(c(200,200), c("mm","mm")),
  heights = unit(c(200,200,200), c("mm","mm","mm")),
  filename = "results/different-treatments-within-varieties/venn-diagrams.pdf",
  config = list(
    list(
      title="Duke of York ubiquitous up-regulated",
      x=c("DEG.Duke.Out.UP", "DEG.Duke.Inf.UP"),
      categories=c("Duke Outside Infected\nUp (vs Mock)", "Duke Inside Infected\nUp (vs Mock)")
    ),
    list(
      title="Duke of York ubiquitous down-regulated",
      x=c("DEG.Duke.Out.DOWN", "DEG.Duke.Inf.DOWN"),
      categories=c("Duke Outside Infected\nDown (vs Mock)", "Duke Inside Infected\nDown (vs Mock)")
    ),
    list(
      title="Sarpo Mira ubiquitous up-regulated",
      x=c("DEG.SMira.Out.UP", "DEG.SMira.Inf.UP"),
      categories=c("S.Mira Outside Infected\nUp (vs Mock)", "S.Mira Inside Infected\nUp (vs Mock)")
    ),
    list(
      title="Sarpo Mira ubiquitous down-regulated",
      x=c("DEG.SMira.Out.DOWN", "DEG.SMira.Inf.DOWN"),
      categories=c("S.Mira Outside Infected\nDown (vs Mock)", "S.Mira Inside Infected\nDown (vs Mock)")
    ),
    list(
      title="Sarpo Shona ubiquitous up-regulated",
      x=c("DEG.SShona.Out.UP", "DEG.SShona.Inf.UP"),
      categories=c("S.Shona Outside Infected\nUp (vs Mock)", "S.Shona Inside Infected\nUp (vs Mock)")
    ),
    list(
      title="Sarpo Shona ubiquitous down-regulated",
      x=c("DEG.SShona.Out.DOWN", "DEG.SShona.Inf.DOWN"),
      categories=c("S.Shona Outside Infected\nDown (vs Mock)", "S.Shona Inside Infected\nDown (vs Mock)")
    )
  )
)
```

## Data frames

### Treatment response

```{r}
for (name in names(up_down_degs)) {
  if (!grepl("_vs_", name) && !grepl("DEG_p", name)) {
    write.csv(up_down_degs[[name]], file = paste("results", "different-treatments-within-varieties", paste(name, "csv", sep = "."), sep = "/"))
    
    genes <- as.data.frame(up_down_degs[[name]]$gene)
    names(genes) <- c("gene")
    annotations <- go_and_func_enrichment(genes)
    write.csv(annotations, file = paste("results", "different-treatments-within-varieties", paste(paste(name, "annotations", sep = "_"), 'csv', sep = "."), sep = "/"))
    
    genes_and_names <- unique(annotations %>% dplyr::select(gene, name))
    write.csv(genes_and_names, file = paste("results", "different-treatments-within-varieties", paste(paste(name, "func_ann_names", sep = "_"), 'csv', sep = "."), sep = "/"))
  }
}
```

### Ubiquitous response to infection

```{r}
for (variety in names(varieties)) {
  for (regulation_direction in regulation_directions) {
    out_name = paste("DEG", variety, "Out", regulation_direction, sep = ".")
    inf_name = paste("DEG", variety, "Inf", regulation_direction, sep = ".")
    name = paste(out_name, inf_name, sep = "_")
    
    df = merge(up_down_degs[[out_name]], up_down_degs[[inf_name]], by = "gene", suffixes = c("_Out", "_Inf"))
    write.csv(df, file = paste("results", "different-treatments-within-varieties", paste(name, "csv", sep = "."), sep = "/"))
    
    genes <- as.data.frame(df$gene)
    names(genes) <- c("gene")
    annotations <- go_and_func_enrichment(genes)
    write.csv(annotations, file = paste("results", "different-treatments-within-varieties", paste(paste(name, "annotations", sep = "_"), 'csv', sep = "."), sep = "/"))
    
    genes_and_names <- unique(annotations %>% dplyr::select(gene, name))
    write.csv(genes_and_names, file = paste("results", "different-treatments-within-varieties", paste(paste(name, "func_ann_names", sep = "_"), 'csv', sep = "."), sep = "/"))
  }
}
```

# Comparing the same treatment BETWEEN varieties

## Volcano plots

```{r, echo=FALSE, fig.height=20, fig.width=14}
volcano_plots_from_comparisons(res,
                               filename = "results/same-treatment-between-varieties/volcano-plots.pdf",
                               ncol = 2,
                               nrow = 3,
                               widths = unit(c(160,160), c("mm","mm")),
                               heights = unit(c(150,150,150), c("mm","mm","mm")),
                               condition = function(name) { grepl("_vs_", name) })
```

## Venn diagrams

### Treatment response (3-way)

```{r, echo=FALSE, fig.height=13, fig.width=13}
create_venn_diagrams_grid(
  up_down_degs,
  ncol = 2,
  nrow = 2,
  widths = unit(c(200, 200), c("mm", "mm")),
  heights = unit(c(200, 200), c("mm", "mm")),
  filename = "results/same-treatment-between-varieties/treatment-response-venn-diagrams.pdf",
  config = list(
    list(
      title="Genes up-regulated in out of infected areas",
      x=c("DEG.Duke.Out.UP", "DEG.SMira.Out.UP", "DEG.SShona.Out.UP"),
      categories=c("Duke Out Up", "S.Mira Out Up", "S.Shona Out Up")
    ),
    list(
      title="Genes down-regulated in out of infected areas",
      x=c("DEG.Duke.Out.DOWN", "DEG.SMira.Out.DOWN", "DEG.SShona.Out.DOWN"),
      categories=c("Duke Out Down", "S.Mira Out Down", "S.Shona Out Down")
    ),
    list(
      title="Genes up-regulated in infected areas",
      x=c("DEG.Duke.Inf.UP", "DEG.SMira.Inf.UP", "DEG.SShona.Inf.UP"),
      categories=c("Duke Inf Up", "S.Mira Inf Up", "S.Shona Inf Up")
    ),
    list(
      title="Genes down-regulated in infected areas",
      x=c("DEG.Duke.Inf.DOWN", "DEG.SMira.Inf.DOWN", "DEG.SShona.Inf.DOWN"),
      categories=c("Duke Inf Down", "S.Mira Inf Down", "S.Shona Inf Down")
    )
  )
)
```

### Sarpo shared baseline differential expression

```{r, echo=FALSE, fig.height=13, fig.width=13}
create_venn_diagrams_grid(
  up_down_degs,
  ncol = 1,
  nrow = 2,
  widths = unit(c(200), c("mm")),
  heights = unit(c(200,200), c("mm","mm")),
  filename = "results/same-treatment-between-varieties/venn-diagrams.pdf",
  config = list(
    list(
      title="Genes upregulated in Sarpo controls (compared with Duke controls)",
      x=c("DEG.SMira_vs_Duke.Control.UP", "DEG.SShona_vs_Duke.Control.UP"),
      categories=c("Mira Control vs\nDuke Control Up", "Shona Control vs\nDuke Control Up")
    ),
    list(
      title="Genes downregulated in Sarpo controls (compared with Duke controls)",
      x=c("DEG.SMira_vs_Duke.Control.DOWN", "DEG.SShona_vs_Duke.Control.DOWN"),
      categories=c("Mira Control vs\nDuke Control Down", "Shona Control vs\nDuke Control Down")
    )
  )
)
```

## Data frames

### Variety-specific treatment response

```{r}
for (name in names(up_down_degs)) {
  if (grepl("_vs_", name) && !grepl("DEG_p", name)) {
    write.csv(up_down_degs[[name]], file = paste("results", "same-treatment-between-varieties", paste(name, "csv", sep = "."), sep = "/"))
    
    genes <- as.data.frame(up_down_degs[[name]]$gene)
    names(genes) <- c("gene")
    annotations <- go_and_func_enrichment(genes)
    write.csv(annotations, file = paste("results", "same-treatment-between-varieties", paste(paste(name, "annotations", sep = "_"), 'csv', sep = "."), sep = "/"))
    
    genes_and_names <- unique(annotations %>% dplyr::select(gene, name))
    write.csv(genes_and_names, file = paste("results", "same-treatment-between-varieties", paste(paste(name, "func_ann_names", sep = "_"), 'csv', sep = "."), sep = "/"))
  }
}
```

### Constitutive UP

```{r}
shona_constitutive_up <- merge(up_down_degs$DEG.SShona_vs_Duke.Control.UP, up_down_degs$DEG.SShona_vs_Duke.Inf.UP, by = "gene", suffixes = c("_Control", "_Inf"))
shona_constitutive_up_annotated_genes <- go_and_func_enrichment(
  in_intersection(up_down_degs$DEG.SShona_vs_Duke.Control.UP$gene, up_down_degs$DEG.SShona_vs_Duke.Inf.UP$gene)
)

mira_constitutive_up <- merge(up_down_degs$DEG.SMira_vs_Duke.Control.UP, up_down_degs$DEG.SMira_vs_Duke.Inf.UP, by = "gene", suffixes = c("_Control", "_Inf"))
mira_constitutive_up_annotated_genes <- go_and_func_enrichment(
  in_intersection(up_down_degs$DEG.SMira_vs_Duke.Control.UP$gene, up_down_degs$DEG.SMira_vs_Duke.Inf.UP$gene)
)

tolerant_constitutive_up <- merge(shona_constitutive_up, mira_constitutive_up, by = "gene", suffixes = c ("_Shona_vs_Duke", "_Mira_vs_Duke"))
tolerant_constitutive_up_genes <- c(tolerant_constitutive_up$gene)
tolerant_constitutive_up_annotated_genes <- mira_constitutive_up_annotated_genes[mira_constitutive_up_annotated_genes$gene %in% tolerant_constitutive_up_genes,]

write.csv(shona_constitutive_up, file = "results/same-treatment-between-varieties/constitutive-shona-upregulation.csv")
write.csv(shona_constitutive_up_annotated_genes, file = "results/same-treatment-between-varieties/constitutive-shona-upregulation-annotations.csv")
write.csv(mira_constitutive_up, file = "results/same-treatment-between-varieties/constitutive-mira-upregulation.csv")
write.csv(mira_constitutive_up_annotated_genes, file = "results/same-treatment-between-varieties/constitutive-mira-upregulation-annotations.csv")
write.csv(tolerant_constitutive_up, file = 
"results/same-treatment-between-varieties/tolerant-constitutive-upregulation.csv")
write.csv(tolerant_constitutive_up_annotated_genes, file = "results/same-treatment-between-varieties/tolerant-constitutive-upregulation-annotations.csv")
```

# Selection of genes for investigation

```{r}
prgs <- readLines('results/prgs')
```

# Heatmaps


```{r}
source('./heatmap.R')

if (!dir.exists('results/heatmaps')) {
  dir.create('results/heatmaps')
}

heatmap_definition <- list(
  inf = list(
    name = 'Inf. vs Mock',
    comparisons = c('DEG.Duke.Inf', 'DEG.SMira.Inf', 'DEG.SShona.Inf'),
    suffixes = c('_Duke', '_Mira', '_Shona')
  ),
  outinf = list(
    name = 'OutInf. vs Mock',
    comparisons = c('DEG.Duke.Out', 'DEG.SMira.Out', 'DEG.SShona.Out'),
    suffixes = c('_Duke', '_Mira', '_Shona')
  ),
  # control = list(
  #   name = 'Control vs Mock',
  #   comparisons = c('DEG.Duke.Control', 'DEG.SMira.Control', 'DEG.SShona.Control'),
  #   suffixes = c('_Duke', '_Mira', '_Shona')
  # ),
  mira_vs_duke = list(
    name = 'S.Mira vs Duke',
    comparisons = c('DEG.SMira_vs_Duke.Control', 'DEG.SMira_vs_Duke.Mock', 'DEG.SMira_vs_Duke.Inf', 'DEG.SMira_vs_Duke.Out'),
    suffixes = c('_Control', '_Mock', '_Inf', '_OutInf')
  ),
  shona_vs_duke = list(
    name = 'S.Shona vs Duke',
    comparisons = c('DEG.SShona_vs_Duke.Control', 'DEG.SShona_vs_Duke.Mock', 'DEG.SShona_vs_Duke.Inf', 'DEG.SShona_vs_Duke.Out'),
    suffixes = c('_Control', '_Mock', '_Inf', '_OutInf')
  )
)

mock_vs_inf_filter <- function(x) {
  rowSums(!is.na(select(x, ends_with('_inf')))) >= 1
}

constitutive_filter <- function(x) {
  rowSums(!is.na(dplyr::select(x, ends_with('mira_vs_duke')))) == 4 | rowSums(!is.na(dplyr::select(x, ends_with('shona_vs_duke')))) == 4
}

# Phosphorylation
# ------------------------------------------------------------------------------
go_terms_phosphorylation <- c(
  'GO:0006468',
  'GO:0004672',
  'GO:0016301',
  'GO:1016310',
  'GO:0106311',
  'GO:0004674',
  'GO:0080163',
  'GO:0000160',
  'GO:0016310',
  'GO:0004675',
  'GO:0033612',
  'GO:0004864'
)

# GO:0080163 - regulation of protein serine/threonine phosphatase activity
# GO:0004675 - transmembrane receptor protein serine/threonine kinase activity
# GO:0106311 - protein threonine kinase activity
# GO:0033612 - receptor serine/threonine kinase binding
# GO:0004674 - protein serine/threonine kinas activity
# GO:0016301 - kinase activity
# GO:0004672 - protein kinase activity
# GO:0006468 - protein phosphorylation
# GO:0016310 - phosphorylation
# GO:0000160 - phosphorelay signal transduction system
# GO:0004864 - protein phosphatase inhibitor activity


go_terms_phosphorylation_order <- c("GO:0080163", "GO:0004675", "GO:0106311", "GO:0033612", "GO:0004674", "GO:0016301", "GO:0004672", "GO:0006468", "GO:0016310", "GO:0000160", "GO:0004864")

plot_heatmap(
  heatmap_definition = heatmap_definition,
  annotated_degs = annotated_degs,
  comparison_list = go_terms_phosphorylation,
  comparison_list_order = go_terms_phosphorylation_order,
  filter_by = 'GOterm',
  filter_fn = mock_vs_inf_filter,
  go_annotations = go_annotations,
  filename = 'results/heatmaps/phosphorylation_to_analyse.pdf'
)

plot_heatmap(
  heatmap_definition = heatmap_definition,
  annotated_degs = annotated_degs,
  comparison_list = go_terms_phosphorylation,
  comparison_list_order = go_terms_phosphorylation_order,
  filter_by = 'GOterm',
  filter_fn = mock_vs_inf_filter,
  go_annotations = go_annotations,
  label_rows = FALSE,
  filename = 'results/heatmaps/phosphorylation_to_analyse_no_label.pdf'
)

plot_heatmap(
  heatmap_definition = heatmap_definition,
  annotated_degs = annotated_degs,
  comparison_list = go_terms_phosphorylation,
  comparison_list_order = go_terms_phosphorylation_order,
  filter_by = 'GOterm',
  filter_fn = constitutive_filter,
  go_annotations = go_annotations,
  filename = 'results/heatmaps/phosphorylation_to_analyse_constitutive.pdf'
)

plot_heatmap(
  heatmap_definition = heatmap_definition,
  annotated_degs = annotated_degs,
  comparison_list = go_terms_phosphorylation,
  comparison_list_order = go_terms_phosphorylation_order,
  filter_by = 'GOterm',
  filter_fn = constitutive_filter,
  go_annotations = go_annotations,
  label_rows = FALSE,
  filename = 'results/heatmaps/phosphorylation_to_analyse_constitutive_no_label.pdf'
)

# Hormones
# ------------------------------------------------------------------------------
go_terms_hormones <- c(
  "GO:0009737",
  "GO:0009738",
  "GO:0009873",
  "GO:2000022",
  "GO:0009753",
  "GO:0009755",
  "GO:0009695",
  "GO:0009688",
  "GO:0009694",
  "GO:0052694"
)

go_terms_hormones_order <- c(
  "GO:0009688",
  "GO:0009738",
  "GO:0009737",
  "GO:0009695",
  "GO:0009694",
  "GO:2000022",
  "GO:0009753",
  "GO:0052694",
  "GO:0009755",
  "GO:0009873"
)

# GO:0009688 - abscisic acid biosynthetic process
# GO:0009738 - abscisic acid-activated signaling pathway
# GO:0009737 - response to abscisic acid
# GO:0009695 - jasmonic acid biosynthetic process
# GO:0009694 - jasmonic acid metabolic process
# GO:2000022 - regulation of jasmonic acid mediated signaling pathway
# GO:0009753 - response to jasmonic acid
# GO:0052694 - jasmonoyl-isoleucine-12-hydroxylase activity
# GO:0009755 - hormone-mediated signaling pathway
# GO:0009873 - ethylene-activated signaling pathway

plot_heatmap(
  heatmap_definition = heatmap_definition,
  annotated_degs = annotated_degs,
  comparison_list = go_terms_hormones,
  comparison_list_order = go_terms_hormones_order,
  filter_by = 'GOterm',
  filter_fn = mock_vs_inf_filter,
  go_annotations = go_annotations,
  filename = 'results/heatmaps/hormones_to_analyse.pdf'
)

plot_heatmap(
  heatmap_definition = heatmap_definition,
  annotated_degs = annotated_degs,
  comparison_list = go_terms_hormones,
  comparison_list_order = go_terms_hormones_order,
  filter_by = 'GOterm',
  filter_fn = mock_vs_inf_filter,
  go_annotations = go_annotations,
  label_rows = FALSE,
  filename = 'results/heatmaps/hormones_to_analyse_no_label.pdf'
)

plot_heatmap(
  heatmap_definition = heatmap_definition,
  annotated_degs = annotated_degs,
  comparison_list = go_terms_hormones,
  comparison_list_order = go_terms_hormones_order,
  filter_by = 'GOterm',
  filter_fn = constitutive_filter,
  go_annotations = go_annotations,
  filename = 'results/heatmaps/hormones_to_analyse_constitutive.pdf'
)

plot_heatmap(
  heatmap_definition = heatmap_definition,
  annotated_degs = annotated_degs,
  comparison_list = go_terms_hormones,
  comparison_list_order = go_terms_hormones_order,
  filter_by = 'GOterm',
  filter_fn = constitutive_filter,
  go_annotations = go_annotations,
  label_rows = FALSE,
  filename = 'results/heatmaps/hormones_to_analyse_constitutive_no_label.pdf'
)

# Example using gene labels
# ------------------------------------------------------------------------------
# TODO: change this
gene_labels <- head(unique(annotated_degs$DEG.Duke.Inf$gene), 100)

plot_heatmap(
  heatmap_definition,
  annotated_degs,
  gene_labels,
  filter_by = 'gene',
  filter_fn = mock_vs_inf_filter,
  filename = 'results/heatmaps/test_with_gene_labels.pdf'
)

# R genes
# ------------------------------------------------------------------------------
plot_heatmap(
  heatmap_definition = heatmap_definition,
  annotated_degs = annotated_degs,
  comparison_list = prgs,
  filter_by = 'transcript_id',
  filter_fn = constitutive_filter,
  filename = 'results/heatmaps/rgenes.pdf'
)
```

# Bar charts

```{r}
if (!dir.exists('results/bar-charts')) {
  dir.create('results/bar-charts')
}

bup_data <- data.frame(
  values = c(
    nrow(up_down_degs$DEG.SMira_vs_Duke.Control.UP),
    nrow(up_down_degs$DEG.SShona_vs_Duke.Control.UP),
    length(intersect(up_down_degs$DEG.SMira_vs_Duke.Control.UP$gene, up_down_degs$DEG.SShona_vs_Duke.Control.UP$gene)),
    
    nrow(up_down_degs$DEG.SMira_vs_Duke.Mock.UP),
    nrow(up_down_degs$DEG.SShona_vs_Duke.Mock.UP),
    length(intersect(up_down_degs$DEG.SMira_vs_Duke.Mock.UP$gene, up_down_degs$DEG.SShona_vs_Duke.Mock.UP$gene)),
    
    nrow(up_down_degs$DEG.SMira_vs_Duke.Inf.UP),
    nrow(up_down_degs$DEG.SShona_vs_Duke.Inf.UP),
    length(intersect(up_down_degs$DEG.SMira_vs_Duke.Inf.UP$gene, up_down_degs$DEG.SShona_vs_Duke.Inf.UP$gene)),
    
    nrow(up_down_degs$DEG.SMira_vs_Duke.Out.UP),
    nrow(up_down_degs$DEG.SShona_vs_Duke.Out.UP),
    length(intersect(up_down_degs$DEG.SMira_vs_Duke.Out.UP$gene, up_down_degs$DEG.SShona_vs_Duke.Out.UP$gene)),
    
    length(Reduce(intersect, list(up_down_degs$DEG.SMira_vs_Duke.Control.UP$gene, up_down_degs$DEG.SMira_vs_Duke.Mock.UP$gene, up_down_degs$DEG.SMira_vs_Duke.Inf.UP$gene, up_down_degs$DEG.SMira_vs_Duke.Out.UP$gene))),
    length(Reduce(intersect, list(up_down_degs$DEG.SShona_vs_Duke.Control.UP$gene, up_down_degs$DEG.SShona_vs_Duke.Mock.UP$gene, up_down_degs$DEG.SShona_vs_Duke.Inf.UP$gene, up_down_degs$DEG.SShona_vs_Duke.Out.UP$gene))),
    length(Reduce(intersect, list(up_down_degs$DEG.SMira_vs_Duke.Control.UP$gene, up_down_degs$DEG.SMira_vs_Duke.Mock.UP$gene, up_down_degs$DEG.SMira_vs_Duke.Inf.UP$gene, up_down_degs$DEG.SMira_vs_Duke.Out.UP$gene, up_down_degs$DEG.SShona_vs_Duke.Control.UP$gene, up_down_degs$DEG.SShona_vs_Duke.Mock.UP$gene, up_down_degs$DEG.SShona_vs_Duke.Inf.UP$gene, up_down_degs$DEG.SShona_vs_Duke.Out.UP$gene)))
  ),
  group = factor(
    rep(c('Control', 'Mock', 'Infected area', 'Outside infected\narea', 'In all treatments'), each = 3),
    levels = c('Control', 'Mock', 'Infected area', 'Outside infected\narea', 'In all treatments')
  ),
  subgroup = factor(
    rep(c('Sarpo Mira vs Duke', 'Sarpo Shona vs Duke', 'In both Sarpos'), 5),
    levels = c('Sarpo Mira vs Duke', 'Sarpo Shona vs Duke', 'In both Sarpos')
  )
)

p <- ggplot(bup_data,
       aes(x = group,
           y = values,
           fill = subgroup)) +
  geom_col(position = position_dodge(0.6), width = 0.5) +
  geom_text(aes(label = values, group = subgroup), position = position_dodge(0.6), vjust = -0.5, size = 2.5) +
  theme_bw() +
  theme(
    axis.title.x=element_blank(),
    axis.title.y=element_blank(),
    panel.grid.major.x = element_blank(),
    legend.title= element_blank()
  )
ggsave(filename = 'results/bar-charts/up.png', width = 10, height = 6)





bdown_data <- data.frame(
  values = c(
    nrow(up_down_degs$DEG.SMira_vs_Duke.Control.DOWN),
    nrow(up_down_degs$DEG.SShona_vs_Duke.Control.DOWN),
    length(intersect(up_down_degs$DEG.SMira_vs_Duke.Control.DOWN$gene, up_down_degs$DEG.SShona_vs_Duke.Control.DOWN$gene)),
    
    nrow(up_down_degs$DEG.SMira_vs_Duke.Mock.DOWN),
    nrow(up_down_degs$DEG.SShona_vs_Duke.Mock.DOWN),
    length(intersect(up_down_degs$DEG.SMira_vs_Duke.Mock.DOWN$gene, up_down_degs$DEG.SShona_vs_Duke.Mock.DOWN$gene)),
    
    nrow(up_down_degs$DEG.SMira_vs_Duke.Inf.DOWN),
    nrow(up_down_degs$DEG.SShona_vs_Duke.Inf.DOWN),
    length(intersect(up_down_degs$DEG.SMira_vs_Duke.Inf.DOWN$gene, up_down_degs$DEG.SShona_vs_Duke.Inf.DOWN$gene)),
    
    nrow(up_down_degs$DEG.SMira_vs_Duke.Out.DOWN),
    nrow(up_down_degs$DEG.SShona_vs_Duke.Out.DOWN),
    length(intersect(up_down_degs$DEG.SMira_vs_Duke.Out.DOWN$gene, up_down_degs$DEG.SShona_vs_Duke.Out.DOWN$gene)),
    
    length(Reduce(intersect, list(up_down_degs$DEG.SMira_vs_Duke.Control.DOWN$gene, up_down_degs$DEG.SMira_vs_Duke.Mock.DOWN$gene, up_down_degs$DEG.SMira_vs_Duke.Inf.DOWN$gene, up_down_degs$DEG.SMira_vs_Duke.Out.DOWN$gene))),
    length(Reduce(intersect, list(up_down_degs$DEG.SShona_vs_Duke.Control.DOWN$gene, up_down_degs$DEG.SShona_vs_Duke.Mock.DOWN$gene, up_down_degs$DEG.SShona_vs_Duke.Inf.DOWN$gene, up_down_degs$DEG.SShona_vs_Duke.Out.DOWN$gene))),
    length(Reduce(intersect, list(up_down_degs$DEG.SMira_vs_Duke.Control.DOWN$gene, up_down_degs$DEG.SMira_vs_Duke.Mock.DOWN$gene, up_down_degs$DEG.SMira_vs_Duke.Inf.DOWN$gene, up_down_degs$DEG.SMira_vs_Duke.Out.DOWN$gene, up_down_degs$DEG.SShona_vs_Duke.Control.DOWN$gene, up_down_degs$DEG.SShona_vs_Duke.Mock.DOWN$gene, up_down_degs$DEG.SShona_vs_Duke.Inf.DOWN$gene, up_down_degs$DEG.SShona_vs_Duke.Out.DOWN$gene)))
  ),
  group = factor(
    rep(c('Control', 'Mock', 'Infected area', 'Outside infected\narea', 'In all treatments'), each = 3),
    levels = c('Control', 'Mock', 'Infected area', 'Outside infected\narea', 'In all treatments')
  ),
  subgroup = factor(
    rep(c('Sarpo Mira vs Duke', 'Sarpo Shona vs Duke', 'In both Sarpos'), 5),
    levels = c('Sarpo Mira vs Duke', 'Sarpo Shona vs Duke', 'In both Sarpos')
  )
)

p <- ggplot(bdown_data,
       aes(x = group,
           y = values,
           fill = subgroup)) +
  geom_col(position = position_dodge(0.6), width = 0.5) +
  geom_text(aes(label = values, group = subgroup), position = position_dodge(0.6), vjust = -0.5, size = 2.5) +
  theme_bw() +
  theme(
    axis.title.x=element_blank(),
    axis.title.y=element_blank(),
    panel.grid.major.x = element_blank(),
    legend.title= element_blank()
  )
ggsave(filename = 'results/bar-charts/down.png', width = 10, height = 6)
```

# Plotting the normalized gene counts across the different time points.

It is a graph for an individual gene.

```{r, echo=FALSE, fig.height=5, fig.width=20}
plotCounts(dds_Blight, gene= "Soltu.DM.01G027350", intgroup=c("variety","treatment"))
```

# Selection of top 20 genes

NOTE: this section can be automated.

## Hold-out Duke

```{r}
hold_out <- up_down_degs$DEG.Duke.Inf.UP
hold_out <- hold_out[order(hold_out$padj), ]
hold_out <- hold_out[1:20, ]

hold_out_with_variety1 <- merge(hold_out, up_down_degs$DEG.SMira.Inf.UP, by = "gene", all.x = T, suffixes = c("_Duke", "_SMira"))
hold_out_with_variety2 <- merge(hold_out, up_down_degs$DEG.SShona.Inf.UP, by = "gene", all.x = T, suffixes = c("_Duke", "_SShona"))
top20 <- merge(hold_out_with_variety1, hold_out_with_variety2)
top20 <- top20[order(top20$padj_Duke), ]
rownames(top20) <- NULL

top20_annotations <- go_and_func_enrichment(as.data.frame(table(top20$gene, dnn = c("gene"))))

top20_genes_and_names <- unique(top20_annotations %>% dplyr::select(gene, name))

write.csv(top20, "results/duke_inf_top20.csv")
write.csv(top20_annotations, "results/duke_inf_top20_annotations.csv")
write.csv(top20_genes_and_names, "results/duke_inf_top20_func_ann_names.csv")
```

## Hold-out S.Mira

```{r}
hold_out <- up_down_degs$DEG.SMira.Inf.UP
hold_out <- hold_out[order(hold_out$padj), ]
hold_out <- hold_out[1:20, ]

hold_out_with_variety1 <- merge(hold_out, up_down_degs$DEG.Duke.Inf.UP, by = "gene", all.x = T, suffixes = c("_SMira", "_Duke"))
hold_out_with_variety2 <- merge(hold_out, up_down_degs$DEG.SShona.Inf.UP, by = "gene", all.x = T, suffixes = c("_SMira", "_SShona"))
top20 <- merge(hold_out_with_variety1, hold_out_with_variety2)
top20 <- top20[order(top20$padj_SMira), ]
rownames(top20) <- NULL

top20_annotations <- go_and_func_enrichment(as.data.frame(table(top20$gene, dnn = c("gene"))))

top20_genes_and_names <- unique(top20_annotations %>% dplyr::select(gene, name))

write.csv(top20, "results/smira_inf_top20.csv")
write.csv(top20_annotations, "results/smira_inf_top20_annotations.csv")
write.csv(top20_genes_and_names, "results/smira_inf_top20_func_ann_names.csv")
```

## Hold-out S.Shona

```{r}
hold_out <- up_down_degs$DEG.SShona.Inf.UP
hold_out <- hold_out[order(hold_out$padj), ]
hold_out <- hold_out[1:20, ]

hold_out_with_variety1 <- merge(hold_out, up_down_degs$DEG.Duke.Inf.UP, by = "gene", all.x = T, suffixes = c("_SShona", "_Duke"))
hold_out_with_variety2 <- merge(hold_out, up_down_degs$DEG.SMira.Inf.UP, by = "gene", all.x = T, suffixes = c("_SShona", "_SMira"))
top20 <- merge(hold_out_with_variety1, hold_out_with_variety2)
top20 <- top20[order(top20$padj_SShona), ]
rownames(top20) <- NULL

top20_annotations <- go_and_func_enrichment(as.data.frame(table(top20$gene, dnn = c("gene"))))

top20_genes_and_names <- unique(top20_annotations %>% dplyr::select(gene, name))


write.csv(top20, "results/sshona_inf_top20.csv")
write.csv(top20_annotations, "results/sshona_inf_top20_annotations.csv")
write.csv(top20_genes_and_names, "results/sshona_inf_top20_func_ann_names.csv")
```

# Save all data

```{r}
save(res, degs, annotated_degs, up_down_degs, annotated_up_down_degs, file = "results/deseq2_analysis.RData")
```
